{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#### Start in KI_Compartment root folder\n",
    "# You must have SymITS2 and all its dependencies installed and working from your home folder (i.e. at ~/SymITS2)\n",
    "# You must have a working version of QIIME, with UCLUST and USEARCH61 installed.\n",
    "# You must have the nucleotide NCBI blast database locally on your machine, and a working version of blast\n",
    "    # You might want to put this on a hard drive, as it is currently ~94GB of data (as of August 1, 2017)\n",
    "    # Download Blastdb from ftp://ftp.ncbi.nlm.nih.gov/blast/documents/blastdb.html, dataset: nt.*tar.gz.\n",
    "    # Then make sure to unzip all folders and place in one folder called \"blastdb\"\n",
    "    # Mine is at \"/media/danielle/CoralDrive/blastdb\"\n",
    "    # For getting blast installed on Linux Ubuntu, this website was helpful: http://www.blopig.com/blog/2014/04/quick-standalone-blast-setup-for-ubuntu-linux/\n",
    "# You must have an updated version of R, I used R 3.4.1 (2017-06-30, Single Candle)\n",
    "    # Required R packages are specified in SymITS2\n",
    "\n",
    "# Prep raw data\n",
    "#chmod u+x data/Bioinf/extract_unzip_prep_rawdata.sh # Make sure script can run\n",
    "#data/Bioinf/extract_unzip_prep_rawdata.sh # This does the following:\n",
    "    # Unzips all original files\n",
    "    # Makes configs for Bokulich qc\n",
    "    # Runs Bokulich qc\n",
    "    # Makes configs for Illumina paired end merging\n",
    "    # Merges Illumina pairs, includes Q30 check\n",
    "    # Filters merged reads\n",
    "    \n",
    "# Or run each command from extract_unzip_prep_rawdata individually...\n",
    "find . -type f -exec gunzip {} + # Unzip these files\n",
    "# They are going to get really big! And this may take quite some time!\n",
    "\n",
    "python data/Bioinf/make_configs.py # Make the Illumina-utils config files for the Bokulich QC method\n",
    "\n",
    "python data/Bioinf/boku_qc.py # Run Bokulich QC method on all applicable files\n",
    "chmod u+x data/Bioinf/boku_qc_KI_Compartment.sh # Give permissions so .sh file will run\n",
    "\n",
    "source ~/virtual-envs/illumina-utils-v2.0.0/bin/activate # Activate Illumina-utils\n",
    "data/Bioinf/boku_qc_KI_Compartment.sh # Run .sh file that was just created \n",
    "\n",
    "python data/Bioinf/make_merge_configs.py # Make the Illumina-utils config files for merging pairs\n",
    "\n",
    "python data/Bioinf/iu_merge_pairs.py # Merge pairs using Illumina-utils\n",
    "chmod u+x data/Bioinf/iu-merge-pairs_KI_Compartment.sh # Give permissions so .sh file will run\n",
    "data/Bioinf/iu-merge-pairs_KI_Compartment.sh # Run .sh file that was just created \n",
    "# find . -maxdepth 1 -name # To check how far along the merging is (because it takes a long time to do hundreds of samples)\n",
    "\n",
    "python data/Bioinf/iu_filter_merged_reads.py # Filter merged reads (MAX-MISMATCH=3) using Illumina-utils\n",
    "chmod u+x data/Bioinf/iu-filter-merged-reads_KI_Compartment.sh # Give permissions so .sh file will run\n",
    "data/Bioinf/iu-filter-merged-reads_KI_Compartment.sh # Run .sh file that was just created \n",
    "\n",
    "python data/Bioinf/rename.py # Rename merged files for downstream processing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Now, because I named these samples unneccesarily differently when I sent them in for sequencing, let's fix that before moving downstream...\n",
    "\n",
    "rename 's/DClaar_2014_S_\\.*/KI14SSYM0/' *.fasta\n",
    "rename 's/DClaar_2014_W_\\.*/KI14WSYM0/' *.fasta\n",
    "rename 's/DClaar_2015a_S_\\.*/KI15aSSYM0/' *.fasta\n",
    "rename 's/DClaar_2015a_W_\\.*/KI15aWSYM0/' *.fasta\n",
    "rename 's/DClaar_2015b_W_\\.*/KI15bWSYM0/' *.fasta\n",
    "rename 's/DClaar_2015b_S_\\.*/KI15bSSYM0/' *.fasta\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 1.9.1 with Python 2.7.12\n",
      "Command line parameters: -g GTGAATTGCAGAACTCCGTG -e 0.15 data/fasta/combined_seqs_chimera_filtered.fasta --trimmed-only -o data/fasta/trimF.fasta\n",
      "Trimming 1 adapter with at most 15.0% errors in single-end mode ...\n",
      "Finished in 144.72 s (20 us/read; 3.05 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               7,353,302\n",
      "Reads with adapters:                 7,345,426 (99.9%)\n",
      "Reads written (passing filters):     7,345,426 (99.9%)\n",
      "\n",
      "Total basepairs processed: 2,556,539,985 bp\n",
      "Total written (filtered):  2,407,039,381 bp (94.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGAATTGCAGAACTCCGTG; Type: regular 5'; Length: 20; Trimmed: 7345426 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t12561\t114895.3\t0\t12561\n",
      "7\t1\t448.8\t1\t0 1\n",
      "8\t10\t112.2\t1\t0 10\n",
      "9\t205\t28.1\t1\t143 62\n",
      "10\t26\t7.0\t1\t21 5\n",
      "11\t30\t1.8\t1\t9 21\n",
      "12\t111\t0.4\t1\t61 50\n",
      "13\t271\t0.1\t1\t165 105 1\n",
      "14\t310\t0.0\t2\t58 204 48\n",
      "15\t717\t0.0\t2\t133 258 326\n",
      "16\t1268\t0.0\t2\t161 471 636\n",
      "17\t4286\t0.0\t2\t456 543 1308 1979\n",
      "18\t12046\t0.0\t2\t915 1292 8142 1697\n",
      "19\t298390\t0.0\t2\t7324 260202 27957 2907\n",
      "20\t6999896\t0.0\t3\t5775653 737798 365069 121376\n",
      "21\t11640\t0.0\t3\t1512 7305 2068 755\n",
      "22\t1537\t0.0\t3\t4 5 814 714\n",
      "23\t388\t0.0\t3\t1 4 2 381\n",
      "24\t4\t0.0\t3\t1 2 0 1\n",
      "25\t6\t0.0\t3\t4 1 1\n",
      "26\t10\t0.0\t3\t0 5 1 4\n",
      "27\t8\t0.0\t3\t2 4 0 2\n",
      "28\t4\t0.0\t3\t4\n",
      "29\t4\t0.0\t3\t3 1\n",
      "30\t8\t0.0\t3\t6 0 1 1\n",
      "31\t6\t0.0\t3\t3 3\n",
      "32\t3\t0.0\t3\t1 2\n",
      "33\t5\t0.0\t3\t1 1 3\n",
      "34\t3\t0.0\t3\t1 1 1\n",
      "35\t7\t0.0\t3\t4 2 1\n",
      "36\t5\t0.0\t3\t4 1\n",
      "37\t5\t0.0\t3\t4 0 1\n",
      "38\t6\t0.0\t3\t2 3 1\n",
      "39\t2\t0.0\t3\t1 0 0 1\n",
      "40\t3\t0.0\t3\t2 1\n",
      "41\t6\t0.0\t3\t5 0 1\n",
      "42\t3\t0.0\t3\t3\n",
      "43\t1\t0.0\t3\t1\n",
      "44\t3\t0.0\t3\t2 1\n",
      "45\t3\t0.0\t3\t3\n",
      "46\t1\t0.0\t3\t1\n",
      "47\t2\t0.0\t3\t2\n",
      "48\t5\t0.0\t3\t5\n",
      "49\t2\t0.0\t3\t1 1\n",
      "50\t2\t0.0\t3\t1 1\n",
      "51\t1\t0.0\t3\t1\n",
      "52\t3\t0.0\t3\t3\n",
      "53\t4\t0.0\t3\t2 2\n",
      "55\t1\t0.0\t3\t1\n",
      "56\t1\t0.0\t3\t0 0 0 1\n",
      "57\t2\t0.0\t3\t1 1\n",
      "63\t2\t0.0\t3\t2\n",
      "84\t1\t0.0\t3\t0 0 1\n",
      "116\t1\t0.0\t3\t0 0 1\n",
      "130\t1\t0.0\t3\t0 0 1\n",
      "154\t4\t0.0\t3\t0 0 0 4\n",
      "155\t2\t0.0\t3\t0 0 0 2\n",
      "156\t11\t0.0\t3\t0 0 0 11\n",
      "157\t21\t0.0\t3\t0 18 0 3\n",
      "181\t1\t0.0\t3\t0 1\n",
      "199\t1\t0.0\t3\t0 0 0 1\n",
      "200\t2\t0.0\t3\t0 0 0 2\n",
      "207\t4\t0.0\t3\t3 0 1\n",
      "213\t1\t0.0\t3\t0 1\n",
      "218\t9\t0.0\t3\t0 9\n",
      "219\t16\t0.0\t3\t1 14 1\n",
      "229\t1\t0.0\t3\t0 1\n",
      "243\t1\t0.0\t3\t1\n",
      "244\t1\t0.0\t3\t1\n",
      "246\t1\t0.0\t3\t0 1\n",
      "247\t9\t0.0\t3\t8 0 1\n",
      "249\t3\t0.0\t3\t3\n",
      "250\t60\t0.0\t3\t38 21 1\n",
      "251\t1336\t0.0\t3\t973 363\n",
      "252\t38\t0.0\t3\t29 9\n",
      "253\t22\t0.0\t3\t22\n",
      "254\t26\t0.0\t3\t25 1\n",
      "256\t29\t0.0\t3\t28 1\n",
      "\n",
      "This is cutadapt 1.9.1 with Python 2.7.12\n",
      "Command line parameters: -g GTGAATTGCAGAACTCCGTG -e 0.15 data/fasta/trimF.fasta -o data/fasta/trimF2.fasta\n",
      "Trimming 1 adapter with at most 15.0% errors in single-end mode ...\n",
      "Finished in 177.01 s (24 us/read; 2.49 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               7,345,426\n",
      "Reads with adapters:                     4,633 (0.1%)\n",
      "Reads written (passing filters):     7,345,426 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,407,039,381 bp\n",
      "Total written (filtered):  2,406,594,885 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GTGAATTGCAGAACTCCGTG; Type: regular 5'; Length: 20; Trimmed: 4633 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t52\t114772.3\t0\t52\n",
      "4\t6\t28693.1\t0\t6\n",
      "5\t3\t7173.3\t0\t3\n",
      "6\t6\t1793.3\t0\t3 3\n",
      "7\t8\t448.3\t1\t0 8\n",
      "8\t41\t112.1\t1\t0 41\n",
      "9\t64\t28.0\t1\t55 9\n",
      "10\t77\t7.0\t1\t60 17\n",
      "11\t140\t1.8\t1\t109 31\n",
      "12\t15\t0.4\t1\t2 11 2\n",
      "13\t100\t0.1\t1\t0 80 20\n",
      "14\t673\t0.0\t2\t0 531 142\n",
      "15\t336\t0.0\t2\t0 251 85\n",
      "16\t2\t0.0\t2\t0 2\n",
      "17\t11\t0.0\t2\t8 1 2\n",
      "18\t2\t0.0\t2\t1 1\n",
      "19\t1\t0.0\t2\t0 0 1\n",
      "20\t9\t0.0\t3\t3 3 2 1\n",
      "21\t2\t0.0\t3\t1 1\n",
      "22\t4\t0.0\t3\t3 1\n",
      "23\t1\t0.0\t3\t0 1\n",
      "24\t5\t0.0\t3\t2 2 0 1\n",
      "25\t7\t0.0\t3\t0 5 0 2\n",
      "26\t10\t0.0\t3\t3 7\n",
      "27\t1\t0.0\t3\t1\n",
      "28\t5\t0.0\t3\t4 1\n",
      "29\t2\t0.0\t3\t2\n",
      "30\t3\t0.0\t3\t2 1\n",
      "31\t4\t0.0\t3\t1 2 1\n",
      "32\t3\t0.0\t3\t2 1\n",
      "33\t2\t0.0\t3\t1 1\n",
      "34\t5\t0.0\t3\t2 1 0 2\n",
      "35\t1\t0.0\t3\t0 1\n",
      "36\t1\t0.0\t3\t0 1\n",
      "37\t4\t0.0\t3\t3 1\n",
      "38\t6\t0.0\t3\t1 0 3 2\n",
      "40\t3\t0.0\t3\t0 3\n",
      "41\t1\t0.0\t3\t1\n",
      "42\t1\t0.0\t3\t0 0 1\n",
      "43\t3\t0.0\t3\t3\n",
      "44\t1\t0.0\t3\t0 1\n",
      "45\t4\t0.0\t3\t3 1\n",
      "46\t4\t0.0\t3\t0 3 0 1\n",
      "47\t1\t0.0\t3\t0 1\n",
      "48\t1\t0.0\t3\t1\n",
      "60\t22\t0.0\t3\t16 5 1\n",
      "75\t1\t0.0\t3\t0 1\n",
      "97\t1\t0.0\t3\t0 0 0 1\n",
      "102\t1\t0.0\t3\t0 1\n",
      "123\t1\t0.0\t3\t0 0 1\n",
      "136\t27\t0.0\t3\t0 0 0 27\n",
      "137\t32\t0.0\t3\t0 0 0 32\n",
      "138\t2745\t0.0\t3\t0 0 1 2744\n",
      "139\t20\t0.0\t3\t0 1 0 19\n",
      "154\t1\t0.0\t3\t0 0 1\n",
      "161\t1\t0.0\t3\t0 1\n",
      "176\t1\t0.0\t3\t0 0 1\n",
      "189\t3\t0.0\t3\t0 1 2\n",
      "190\t95\t0.0\t3\t0 0 94 1\n",
      "196\t6\t0.0\t3\t0 6\n",
      "197\t1\t0.0\t3\t1\n",
      "200\t3\t0.0\t3\t0 0 2 1\n",
      "202\t2\t0.0\t3\t0 1 1\n",
      "211\t1\t0.0\t3\t0 1\n",
      "213\t1\t0.0\t3\t0 0 1\n",
      "225\t1\t0.0\t3\t0 0 0 1\n",
      "228\t1\t0.0\t3\t0 1\n",
      "230\t2\t0.0\t3\t0 1 1\n",
      "231\t5\t0.0\t3\t0 0 4 1\n",
      "232\t26\t0.0\t3\t0 0 25 1\n",
      "233\t2\t0.0\t3\t0 1 1\n",
      "\n",
      "This is cutadapt 1.9.1 with Python 2.7.12\n",
      "Command line parameters: -a AAGCATATAAGTAAGCGGAGG -e 0.15 data/fasta/trimF2.fasta --trimmed-only -o data/fasta/trimF2_trimR.fasta\n",
      "Trimming 1 adapter with at most 15.0% errors in single-end mode ...\n",
      "Finished in 138.52 s (19 us/read; 3.18 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               7,345,426\n",
      "Reads with adapters:                 7,332,475 (99.8%)\n",
      "Reads written (passing filters):     7,332,475 (99.8%)\n",
      "\n",
      "Total basepairs processed: 2,406,594,885 bp\n",
      "Total written (filtered):  2,247,881,322 bp (93.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AAGCATATAAGTAAGCGGAGG; Type: regular 3'; Length: 21; Trimmed: 7332475 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20-21 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 0.0%\n",
      "  G: 0.0%\n",
      "  T: 0.0%\n",
      "  none/other: 100.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t7\t114772.3\t0\t7\n",
      "4\t3\t28693.1\t0\t3\n",
      "7\t6\t448.3\t1\t6\n",
      "9\t10\t28.0\t1\t9 1\n",
      "10\t31\t7.0\t1\t16 15\n",
      "11\t76\t1.8\t1\t6 70\n",
      "12\t10\t0.4\t1\t1 9\n",
      "13\t43\t0.1\t1\t30 7 6\n",
      "14\t229\t0.0\t2\t140 49 40\n",
      "15\t1498\t0.0\t2\t1164 105 229\n",
      "16\t2332\t0.0\t2\t473 1593 266\n",
      "17\t4150\t0.0\t2\t1774 1273 1083 20\n",
      "18\t10681\t0.0\t2\t4836 1401 571 3873\n",
      "19\t20705\t0.0\t2\t4204 5119 10379 1003\n",
      "20\t120640\t0.0\t3\t7216 84923 24113 4388\n",
      "21\t7152652\t0.0\t3\t7049737 96907 5442 566\n",
      "22\t18923\t0.0\t3\t2035 14726 1959 203\n",
      "23\t158\t0.0\t3\t6 10 56 86\n",
      "24\t45\t0.0\t3\t9 2 0 34\n",
      "25\t22\t0.0\t3\t8 4 5 5\n",
      "26\t25\t0.0\t3\t8 6 3 8\n",
      "27\t32\t0.0\t3\t14 2 9 7\n",
      "28\t11\t0.0\t3\t4 3 1 3\n",
      "29\t13\t0.0\t3\t4 1 2 6\n",
      "30\t20\t0.0\t3\t7 2 2 9\n",
      "31\t10\t0.0\t3\t6 1 2 1\n",
      "32\t18\t0.0\t3\t10 5 3\n",
      "33\t7\t0.0\t3\t6 0 0 1\n",
      "34\t8\t0.0\t3\t7 0 1\n",
      "35\t8\t0.0\t3\t7 1\n",
      "36\t2\t0.0\t3\t2\n",
      "37\t6\t0.0\t3\t5 0 1\n",
      "38\t5\t0.0\t3\t3 2\n",
      "39\t3\t0.0\t3\t3\n",
      "40\t5\t0.0\t3\t5\n",
      "41\t3\t0.0\t3\t2 1\n",
      "42\t2\t0.0\t3\t2\n",
      "43\t5\t0.0\t3\t1 2 2\n",
      "44\t6\t0.0\t3\t3 1 2\n",
      "45\t2\t0.0\t3\t1 1\n",
      "47\t1\t0.0\t3\t1\n",
      "48\t1\t0.0\t3\t1\n",
      "49\t1\t0.0\t3\t0 1\n",
      "50\t1\t0.0\t3\t1\n",
      "53\t3\t0.0\t3\t3\n",
      "54\t1\t0.0\t3\t0 1\n",
      "55\t2\t0.0\t3\t1 1\n",
      "56\t2\t0.0\t3\t2\n",
      "57\t1\t0.0\t3\t1\n",
      "58\t2\t0.0\t3\t1 0 1\n",
      "59\t1\t0.0\t3\t1\n",
      "61\t1\t0.0\t3\t1\n",
      "62\t1\t0.0\t3\t1\n",
      "63\t2\t0.0\t3\t2\n",
      "68\t3\t0.0\t3\t0 0 3\n",
      "134\t1\t0.0\t3\t0 1\n",
      "136\t13\t0.0\t3\t8 5\n",
      "137\t12\t0.0\t3\t12\n",
      "143\t2\t0.0\t3\t0 0 2\n",
      "144\t1\t0.0\t3\t0 0 1\n",
      "178\t1\t0.0\t3\t1\n",
      "223\t1\t0.0\t3\t0 0 1\n",
      "261\t2\t0.0\t3\t2\n",
      "262\t7\t0.0\t3\t7\n",
      "\n",
      "This is cutadapt 1.9.1 with Python 2.7.12\n",
      "Command line parameters: -a AAGCATATAAGTAAGCGGAGG -e 0.15 data/fasta/trimF2_trimR.fasta -m 250 -o data/fasta/combined_seqs_trimmed.fasta\n",
      "Trimming 1 adapter with at most 15.0% errors in single-end mode ...\n",
      "Finished in 169.29 s (23 us/read; 2.60 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               7,332,475\n",
      "Reads with adapters:                       845 (0.0%)\n",
      "Reads that were too short:                   2 (0.0%)\n",
      "Reads written (passing filters):     7,332,473 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,247,881,322 bp\n",
      "Total written (filtered):  2,247,863,966 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AAGCATATAAGTAAGCGGAGG; Type: regular 3'; Length: 21; Trimmed: 845 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20-21 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 0.0%\n",
      "  G: 0.0%\n",
      "  T: 0.0%\n",
      "  none/other: 100.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t52\t114569.9\t0\t52\n",
      "4\t240\t28642.5\t0\t240\n",
      "5\t95\t7160.6\t0\t95\n",
      "6\t3\t1790.2\t0\t1 2\n",
      "7\t5\t447.5\t1\t2 3\n",
      "8\t24\t111.9\t1\t21 3\n",
      "10\t1\t7.0\t1\t0 1\n",
      "12\t18\t0.4\t1\t16 2\n",
      "16\t1\t0.0\t2\t0 0 1\n",
      "23\t9\t0.0\t3\t0 0 0 9\n",
      "25\t323\t0.0\t3\t0 3 299 21\n",
      "40\t32\t0.0\t3\t0 0 27 5\n",
      "42\t2\t0.0\t3\t0 0 1 1\n",
      "57\t1\t0.0\t3\t0 1\n",
      "115\t32\t0.0\t3\t0 31 1\n",
      "241\t7\t0.0\t3\t0 2 5\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Next, start using SymITS2 by Ross Cunning https://github.com/jrcunning/SymITS2\n",
    "~/SymITS2/qc_trim_reads.sh # This does the following:\n",
    "    # Adds QIIME labels\n",
    "    # Identifies chimeric sequences\n",
    "    # Filters out chimeric sequences\n",
    "    # Runs cutadapt 4 times\n",
    "    # Removes intermediate files from cutadapt process"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "7332473  : data/fasta/combined_seqs_trimmed.fasta (Sequence lengths (mean +/- std): 306.5629 +/- 24.1155)\n",
      "7332473  : Total\n"
     ]
    }
   ],
   "source": [
    "count_seqs.py -i data/fasta/combined_seqs_trimmed.fasta # Count how many seqs there are after quality filtering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "When using programs that use GNU Parallel to process data for publication please cite:\n",
      "\n",
      "  O. Tange (2011): GNU Parallel - The Command-Line Power Tool,\n",
      "  ;login: The USENIX Magazine, February 2011:42-47.\n",
      "\n",
      "This helps funding further development; and it won't cost you a cent.\n",
      "Or you can get GNU Parallel without this requirement by paying 10000 EUR.\n",
      "\n",
      "To silence this citation notice run 'parallel --bibtex' once or use '--no-notice'.\n",
      "\n",
      "When using programs that use GNU Parallel to process data for publication please cite:\n",
      "\n",
      "  O. Tange (2011): GNU Parallel - The Command-Line Power Tool,\n",
      "  ;login: The USENIX Magazine, February 2011:42-47.\n",
      "\n",
      "This helps funding further development; and it won't cost you a cent.\n",
      "Or you can get GNU Parallel without this requirement by paying 10000 EUR.\n",
      "\n",
      "To silence this citation notice run 'parallel --bibtex' once or use '--no-notice'.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# From SymITS2 by Ross Cunning https://github.com/jrcunning/SymITS2\n",
    "# Cluster at 97% within samples (each sample is clustered independently)\n",
    "~/SymITS2/otus_97_bysample.sh data/fasta/combined_seqs_trimmed.fasta data/Bioinf/clust\n",
    "# Arguments are 1) Combined, trimmed sequences (fasta file) and 2)Output directory\n",
    "# Creates data/Bioinf/clust/all_rep_set_rep_set.fasta for downstream use"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 3.4.1 (2017-06-30) -- \"Single Candle\"\n",
      "Copyright (C) 2017 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "  Natural language support but running in an English locale\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> # Get command line arguments\n",
      "> args = commandArgs(trailingOnly=TRUE)\n",
      "> # If two arguments not provided, return an error\n",
      "> if (length(args) < 2) {\n",
      "+   stop(\"must specify query sequences (.fasta) and reference sequences (.fasta)\", call.=FALSE)\n",
      "+ }\n",
      "> \n",
      "> # Import query sequences\n",
      "> library(Biostrings)\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, cbind, colMeans, colnames,\n",
      "    colSums, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, lengths, Map, mapply, match,\n",
      "    mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "Loading required package: stats4\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Loading required package: XVector\n",
      "\n",
      "Attaching package: ‘Biostrings’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    strsplit\n",
      "\n",
      "> otus <- readDNAStringSet(args[1])\n",
      "> names(otus) <- gsub(\" .*$\", \"\", names(otus))\n",
      "> \n",
      "> # Import reference sequences\n",
      "> ref <- readDNAStringSet(args[2])\n",
      "> \n",
      "> # import substitution matrix\n",
      "> EDNAFULL <- matrix(\n",
      "+ strsplit(\n",
      "+ \"5 -4 -4 -4 -4  1  1 -4 -4  1 -4 -1 -1 -1 -2\n",
      "+ -4  5 -4 -4 -4  1 -4  1  1 -4 -1 -4 -1 -1 -2\n",
      "+ -4 -4  5 -4  1 -4  1 -4  1 -4 -1 -1 -4 -1 -2\n",
      "+ -4 -4 -4  5  1 -4 -4  1 -4  1 -1 -1 -1 -4 -2\n",
      "+ -4 -4  1  1 -1 -4 -2 -2 -2 -2 -1 -1 -3 -3 -1\n",
      "+  1  1 -4 -4 -4 -1 -2 -2 -2 -2 -3 -3 -1 -1 -1\n",
      "+  1 -4  1 -4 -2 -2 -1 -4 -2 -2 -3 -1 -3 -1 -1\n",
      "+ -4  1 -4  1 -2 -2 -4 -1 -2 -2 -1 -3 -1 -3 -1\n",
      "+ -4  1  1 -4 -2 -2 -2 -2 -1 -4 -1 -3 -3 -1 -1\n",
      "+  1 -4 -4  1 -2 -2 -2 -2 -4 -1 -3 -1 -1 -3 -1\n",
      "+ -4 -1 -1 -1 -1 -3 -3 -1 -1 -3 -1 -2 -2 -2 -1\n",
      "+ -1 -4 -1 -1 -1 -3 -1 -3 -3 -1 -2 -1 -2 -2 -1\n",
      "+ -1 -1 -4 -1 -3 -1 -3 -1 -3 -1 -2 -2 -1 -2 -1\n",
      "+ -1 -1 -1 -4 -3 -1 -1 -3 -1 -3 -2 -2 -2 -1 -1\n",
      "+ -2 -2 -2 -2 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\",\n",
      "+ split=\" |  |\\n|\\t|\\n\\ \")[[1]],\n",
      "+ nrow=15, byrow=T, \n",
      "+ dimnames=list(c(\"A\",\"T\",\"G\",\"C\",\"S\",\"W\",\"R\",\"Y\",\"K\",\"M\",\"B\", \"V\",\"H\",\"D\",\"N\"),\n",
      "+               c(\"A\",\"T\",\"G\",\"C\",\"S\",\"W\",\"R\",\"Y\",\"K\",\"M\",\"B\", \"V\",\"H\",\"D\",\"N\")))\n",
      "> storage.mode(EDNAFULL) <- \"numeric\"\n",
      "> \n",
      "> # Build function to find which reference sequences produces best alignment score using\n",
      "> # Needleman-Wunsch algorithm with default EMBOSS parameters for DNA sequences.\n",
      "> # -type=\"global-local\" forces whole reference (pattern) sequence to be used, but subject/query may be \n",
      "> # a concescutive subsequence. This is because queries may include slightly longer region of gene \n",
      "> # than references because references were trimmed by o-smart-trim. This means that query sequence \n",
      "> # can have more bases than the reference at beginning or end of the sequence (=end gaps in \n",
      "> # alignment) without penalties.\n",
      "> get.best <- function(x) {\n",
      "+   scores <- pairwiseAlignment(subject=x, pattern=ref, type=\"global-local\", scoreOnly=TRUE,\n",
      "+                            substitutionMatrix=EDNAFULL,\n",
      "+                            gapOpening=10.0, gapExtension=0.5)\n",
      "+   best <- cbind(names(x), names(ref)[which(scores==max(scores))])\n",
      "+   return(best)\n",
      "+ }\n",
      "> \n",
      "> # Set up for parallel computation\n",
      "> library(parallel)\n",
      "> cl <- makeCluster(detectCores())  # Initiate cluster\n",
      "> clusterEvalQ(cl, library(Biostrings))  # Make Biostrings library available to cluster\n",
      "[[1]]\n",
      " [1] \"Biostrings\"   \"XVector\"      \"IRanges\"      \"S4Vectors\"    \"stats4\"      \n",
      " [6] \"BiocGenerics\" \"parallel\"     \"stats\"        \"graphics\"     \"grDevices\"   \n",
      "[11] \"utils\"        \"datasets\"     \"methods\"      \"base\"        \n",
      "\n",
      "[[2]]\n",
      " [1] \"Biostrings\"   \"XVector\"      \"IRanges\"      \"S4Vectors\"    \"stats4\"      \n",
      " [6] \"BiocGenerics\" \"parallel\"     \"stats\"        \"graphics\"     \"grDevices\"   \n",
      "[11] \"utils\"        \"datasets\"     \"methods\"      \"base\"        \n",
      "\n",
      "[[3]]\n",
      " [1] \"Biostrings\"   \"XVector\"      \"IRanges\"      \"S4Vectors\"    \"stats4\"      \n",
      " [6] \"BiocGenerics\" \"parallel\"     \"stats\"        \"graphics\"     \"grDevices\"   \n",
      "[11] \"utils\"        \"datasets\"     \"methods\"      \"base\"        \n",
      "\n",
      "[[4]]\n",
      " [1] \"Biostrings\"   \"XVector\"      \"IRanges\"      \"S4Vectors\"    \"stats4\"      \n",
      " [6] \"BiocGenerics\" \"parallel\"     \"stats\"        \"graphics\"     \"grDevices\"   \n",
      "[11] \"utils\"        \"datasets\"     \"methods\"      \"base\"        \n",
      "\n",
      "> clusterExport(cl, c(\"EDNAFULL\", \"ref\", \"otus\", \"get.best\"))  # Export objects to cluster\n",
      "> \n",
      "> # Run function in parallel to get best hit for each sequence\n",
      "> best <- parSapply(cl, otus, FUN=get.best)\n",
      "> stopCluster(cl)  # Stop cluster\n",
      "> \n",
      "> # Collect results in a data frame\n",
      "> results <- data.frame(otu=rep(names(best), sapply(best, length)),\n",
      "+                       hit=unlist(best), stringsAsFactors = FALSE)\n",
      "> \n",
      "> # Align each sequence with its bets hits from database to get number of mismatches, indels, and % similarity\n",
      "> # Align each otu sequence to its best hits\n",
      "> psa <- pairwiseAlignment(subject=otus[results$otu], pattern=ref[results$hit], type=\"global-local\",\n",
      "+                          substitutionMatrix=EDNAFULL,\n",
      "+                          gapOpening=10.0, gapExtension=0.5)\n",
      "> \n",
      "> # Calculate percent similarity to reference with indels counting as single difference regardless of length\n",
      "> results <- within(results, {\n",
      "+   score <- score(psa)\n",
      "+   len <- width(pattern(psa))\n",
      "+   mis <- nmismatch(psa)\n",
      "+   ins <- unlist(lapply(as.list(insertion(psa)), length))\n",
      "+   del <- unlist(lapply(as.list(deletion(psa)), length))\n",
      "+   sim <- round((len-mis-ins-del)/len, 4) * 100\n",
      "+ })\n",
      "> \n",
      "> # Merge identical hits\n",
      "> results2 <- with(results, aggregate(hit ~ otu + sim + del + ins + mis + len + score, FUN=paste0, collapse=\";\"))\n",
      "> \n",
      "> # Write results to .tsv file\n",
      "> write.table(results2, file=paste(gsub(\"\\\\..*$\", \"\", args[1]), \"_nw_tophits.tsv\", sep=\"\"), \n",
      "+             sep=\"\\t\", quote=FALSE)\n",
      "> \n",
      "> \n"
     ]
    }
   ],
   "source": [
    "# From SymITS2 by Ross Cunning https://github.com/jrcunning/SymITS2\n",
    "# Use global alignment to identify sequences\n",
    "R --vanilla < ~/SymITS2/run_nw.R --args data/Bioinf/clust/all_rep_set_rep_set.fasta data/ITS2db_trimmed_derep.fasta\n",
    "# Arguments are 1) query sequences - rep set fasta file, and 2) reference database sequences in .fasta format\n",
    "# Creates data/Bioinf/clust/all_rep_set_rep_set_nw_tophits.tsv for downstream use"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 3.4.1 (2017-06-30) -- \"Single Candle\"\n",
      "Copyright (C) 2017 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "  Natural language support but running in an English locale\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> library(plyr)\n",
      "> \n",
      "> rm(list=ls())\n",
      "> \n",
      "> meta <- read.csv(\"data/Coralphoto__Metadata/KI_Coralphoto_Metadata_Jan_to_Apr_2017_23March.csv\",header=T)\n",
      "> meta$Year_Pre_Post <- paste(meta$field_season,meta$before.after, sep=\"\")\n",
      "> meta$ref <- paste(meta$Year_Pre_Post,\".tag\",meta$coral_tag, sep=\"\")\n",
      "> meta.forcat <- meta[,c(1:25)]\n",
      "> \n",
      "> map <- read.table(\"data/mapping_file.txt\",stringsAsFactors = FALSE)\n",
      "> colnames(map) <- c(\"SampleID\", \"InputFileName\", \"coral_tag\",\"SampleType\", \"Year\", \"TubeNumber\", \"Coral_Species\",\"Site\",\"Status\",\"Year_Pre_Post\")\n",
      "> \n",
      "> fs2014 <- which(map$Year==2014)\n",
      "> map$Year[fs2014] <- \"KI2014\"\n",
      "> fs2015aa <- which(map$Year_Pre_Post==\"2015Jan_Pre\")\n",
      "> map$Year[fs2015aa] <- \"KI2015a_Pre\"\n",
      "> fs2015ab <- which(map$Year_Pre_Post==\"2015Jan_Post\")\n",
      "> map$Year[fs2015ab] <- \"KI2015a_Post\"\n",
      "> fs2015b <- which(map$Year==\"2015May\")\n",
      "> map$Year[fs2015b] <- \"KI2015b\"\n",
      "> fs2015c <- which(map$Year==\"2015July\")\n",
      "> map$Year[fs2015c] <- \"KI2015c\"\n",
      "> fs2016a <- which(map$Year==\"2016March\")\n",
      "> map$Year[fs2016a] <- \"KI2016a\"\n",
      "> # map$ref <- paste(map$V5,\"FSYM\",map$V3,sep=\"\")\n",
      "> map.compartment <- map\n",
      "> \n",
      "> names(map.compartment)[names(map.compartment)==\"Year\"] <- \"field_season\"\n",
      "> names(map.compartment)[names(map.compartment)==\"Site\"] <- \"site\"\n",
      "> # Sampled the same coral colony twice during one field season\n",
      "> fix <- which(map.compartment$SampleID==\"KI15cFSYM509\")\n",
      "> map.compartment$coral_tag[fix] <- \"341.2\"\n",
      "> \n",
      "> map.compartment$ref <- ifelse(map.compartment$SampleType==\"coral\", map.compartment$ref <- paste(map.compartment$field_season,\".tag\",map.compartment$coral_tag, sep=\"\"), \n",
      "+        ifelse(map.compartment$SampleType==\"water\", map.compartment$ref <- paste(map.compartment$field_season,\".water\",map.compartment$TubeNumber,\".site\",map.compartment$site, sep=\"\"), \n",
      "+               ifelse(map.compartment$SampleType==\"sediment\", map.compartment$ref <- paste(map.compartment$field_season,\".sediment\",map.compartment$TubeNumber,\".site\",map.compartment$site, sep=\"\"), NA)))\n",
      "> \n",
      "> duplicated(map.compartment)\n",
      "  [1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      " [13] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      " [25] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      " [37] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      " [49] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      " [61] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      " [73] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      " [85] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      " [97] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[109] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[121] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[133] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[145] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[157] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[169] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[181] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[193] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[205] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[217] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[229] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[241] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[253] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[265] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[277] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[289] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[301] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[313] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[325] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[337] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[349] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[361] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "[373] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
      "> map.compartment\n",
      "        SampleID      InputFileName coral_tag SampleType field_season\n",
      "1    KI14FSYM001  KI14FSYM001.fasta         1      coral       KI2014\n",
      "2    KI14FSYM002  KI14FSYM002.fasta     2_572      coral       KI2014\n",
      "3    KI14FSYM003  KI14FSYM003.fasta         3      coral       KI2014\n",
      "4    KI14SSYM003  KI14SSYM003.fasta      <NA>   sediment       KI2014\n",
      "5    KI14SSYM004  KI14SSYM004.fasta      <NA>   sediment       KI2014\n",
      "6    KI14FSYM005  KI14FSYM005.fasta         5      coral       KI2014\n",
      "7    KI14SSYM005  KI14SSYM005.fasta      <NA>   sediment       KI2014\n",
      "8    KI14SSYM006  KI14SSYM006.fasta      <NA>   sediment       KI2014\n",
      "9    KI14FSYM007  KI14FSYM007.fasta         7      coral       KI2014\n",
      "10   KI14SSYM007  KI14SSYM007.fasta      <NA>   sediment       KI2014\n",
      "11   KI14FSYM008  KI14FSYM008.fasta         8      coral       KI2014\n",
      "12   KI14SSYM008  KI14SSYM008.fasta      <NA>   sediment       KI2014\n",
      "13   KI14FSYM009  KI14FSYM009.fasta         9      coral       KI2014\n",
      "14   KI14SSYM010  KI14SSYM010.fasta      <NA>   sediment       KI2014\n",
      "15   KI14SSYM011  KI14SSYM011.fasta      <NA>   sediment       KI2014\n",
      "16   KI14FSYM012  KI14FSYM012.fasta    12_635      coral       KI2014\n",
      "17   KI14SSYM012  KI14SSYM012.fasta      <NA>   sediment       KI2014\n",
      "18   KI14FSYM013  KI14FSYM013.fasta        13      coral       KI2014\n",
      "19   KI14SSYM013  KI14SSYM013.fasta      <NA>   sediment       KI2014\n",
      "20   KI14SSYM014  KI14SSYM014.fasta      <NA>   sediment       KI2014\n",
      "21   KI14FSYM015  KI14FSYM015.fasta        15      coral       KI2014\n",
      "22   KI14SSYM015  KI14SSYM015.fasta      <NA>   sediment       KI2014\n",
      "23   KI14SSYM016  KI14SSYM016.fasta      <NA>   sediment       KI2014\n",
      "24   KI14SSYM017  KI14SSYM017.fasta      <NA>   sediment       KI2014\n",
      "25   KI14SSYM018  KI14SSYM018.fasta      <NA>   sediment       KI2014\n",
      "26   KI14SSYM019  KI14SSYM019.fasta      <NA>   sediment       KI2014\n",
      "27   KI14SSYM020  KI14SSYM020.fasta      <NA>   sediment       KI2014\n",
      "28   KI14FSYM041  KI14FSYM041.fasta        41      coral       KI2014\n",
      "29   KI14FSYM043  KI14FSYM043.fasta        43      coral       KI2014\n",
      "30   KI14FSYM044  KI14FSYM044.fasta        44      coral       KI2014\n",
      "31   KI14FSYM050  KI14FSYM050.fasta        50      coral       KI2014\n",
      "32   KI14SSYM050  KI14SSYM050.fasta      <NA>   sediment       KI2014\n",
      "33   KI14SSYM058  KI14SSYM058.fasta      <NA>   sediment       KI2014\n",
      "34   KI14SSYM059  KI14SSYM059.fasta      <NA>   sediment       KI2014\n",
      "35   KI14FSYM060  KI14FSYM060.fasta        60      coral       KI2014\n",
      "36   KI14SSYM060  KI14SSYM060.fasta      <NA>   sediment       KI2014\n",
      "37   KI14SSYM061  KI14SSYM061.fasta      <NA>   sediment       KI2014\n",
      "38   KI14FSYM063  KI14FSYM063.fasta        63      coral       KI2014\n",
      "39   KI14SSYM063  KI14SSYM063.fasta      <NA>   sediment       KI2014\n",
      "40   KI14FSYM069  KI14FSYM069.fasta    69_875      coral       KI2014\n",
      "41   KI14FSYM076  KI14FSYM076.fasta        76      coral       KI2014\n",
      "42   KI14FSYM084  KI14FSYM084.fasta        84      coral       KI2014\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "43   KI14FSYM087  KI14FSYM087.fasta        87      coral       KI2014\n",
      "44   KI14FSYM088  KI14FSYM088.fasta        88      coral       KI2014\n",
      "45   KI14FSYM090  KI14FSYM090.fasta    90_690      coral       KI2014\n",
      "46   KI14FSYM092  KI14FSYM092.fasta    92_860      coral       KI2014\n",
      "47   KI14FSYM094  KI14FSYM094.fasta        94      coral       KI2014\n",
      "48   KI14FSYM097  KI14FSYM097.fasta        97      coral       KI2014\n",
      "49   KI14FSYM098  KI14FSYM098.fasta        98      coral       KI2014\n",
      "50   KI14FSYM102  KI14FSYM102.fasta       102      coral       KI2014\n",
      "51   KI14FSYM105  KI14FSYM105.fasta       105      coral       KI2014\n",
      "52   KI14FSYM106  KI14FSYM106.fasta       106      coral       KI2014\n",
      "53   KI14FSYM107  KI14FSYM107.fasta       107      coral       KI2014\n",
      "54   KI14FSYM109  KI14FSYM109.fasta       109      coral       KI2014\n",
      "55   KI14FSYM126  KI14FSYM126.fasta       126      coral       KI2014\n",
      "56   KI14FSYM129  KI14FSYM129.fasta       129      coral       KI2014\n",
      "57   KI14FSYM131  KI14FSYM131.fasta       131      coral       KI2014\n",
      "58   KI14FSYM132  KI14FSYM132.fasta       132      coral       KI2014\n",
      "59   KI14FSYM133  KI14FSYM133.fasta       133      coral       KI2014\n",
      "60   KI14FSYM134  KI14FSYM134.fasta       134      coral       KI2014\n",
      "61   KI14FSYM135  KI14FSYM135.fasta       135      coral       KI2014\n",
      "62   KI14FSYM136  KI14FSYM136.fasta       136      coral       KI2014\n",
      "63   KI14FSYM137  KI14FSYM137.fasta       137      coral       KI2014\n",
      "64   KI14FSYM140  KI14FSYM140.fasta       140      coral       KI2014\n",
      "65   KI14FSYM142  KI14FSYM142.fasta       142      coral       KI2014\n",
      "66   KI14FSYM143  KI14FSYM143.fasta       143      coral       KI2014\n",
      "67   KI14FSYM144  KI14FSYM144.fasta       144      coral       KI2014\n",
      "68   KI14FSYM147  KI14FSYM147.fasta       147      coral       KI2014\n",
      "69   KI14FSYM149  KI14FSYM149.fasta       149      coral       KI2014\n",
      "70   KI14FSYM151  KI14FSYM151.fasta       151      coral       KI2014\n",
      "71   KI14FSYM265  KI14FSYM265.fasta       265      coral       KI2014\n",
      "72   KI14FSYM385  KI14FSYM385.fasta       385      coral       KI2014\n",
      "73   KI14FSYM387  KI14FSYM387.fasta       387      coral       KI2014\n",
      "74   KI14FSYM448  KI14FSYM448.fasta       448      coral       KI2014\n",
      "75   KI14FSYM451  KI14FSYM451.fasta   451_871      coral       KI2014\n",
      "76   KI14FSYM453  KI14FSYM453.fasta       453      coral       KI2014\n",
      "77   KI14FSYM456  KI14FSYM456.fasta       456      coral       KI2014\n",
      "78   KI14FSYM467  KI14FSYM467.fasta       467      coral       KI2014\n",
      "79   KI14FSYM468  KI14FSYM468.fasta       468      coral       KI2014\n",
      "80   KI14FSYM469  KI14FSYM469.fasta       469      coral       KI2014\n",
      "81   KI14FSYM474  KI14FSYM474.fasta       474      coral       KI2014\n",
      "82   KI14FSYM476  KI14FSYM476.fasta       476      coral       KI2014\n",
      "83   KI14WSYM001  KI14WSYM001.fasta      <NA>      water       KI2014\n",
      "84   KI14WSYM003  KI14WSYM003.fasta      <NA>      water       KI2014\n",
      "85   KI14WSYM004  KI14WSYM004.fasta      <NA>      water       KI2014\n",
      "86   KI14WSYM005  KI14WSYM005.fasta      <NA>      water       KI2014\n",
      "87   KI14WSYM006  KI14WSYM006.fasta      <NA>      water       KI2014\n",
      "88   KI14WSYM007  KI14WSYM007.fasta      <NA>      water       KI2014\n",
      "89   KI14WSYM008  KI14WSYM008.fasta      <NA>      water       KI2014\n",
      "90   KI14WSYM011  KI14WSYM011.fasta      <NA>      water       KI2014\n",
      "91   KI14WSYM012  KI14WSYM012.fasta      <NA>      water       KI2014\n",
      "92   KI14WSYM013  KI14WSYM013.fasta      <NA>      water       KI2014\n",
      "93   KI14WSYM014  KI14WSYM014.fasta      <NA>      water       KI2014\n",
      "94   KI14WSYM015  KI14WSYM015.fasta      <NA>      water       KI2014\n",
      "95   KI14WSYM016  KI14WSYM016.fasta      <NA>      water       KI2014\n",
      "96   KI14WSYM017  KI14WSYM017.fasta      <NA>      water       KI2014\n",
      "97   KI14WSYM021  KI14WSYM021.fasta      <NA>      water       KI2014\n",
      "98   KI14WSYM028  KI14WSYM028.fasta      <NA>      water       KI2014\n",
      "99   KI14WSYM029  KI14WSYM029.fasta      <NA>      water       KI2014\n",
      "100  KI14WSYM030  KI14WSYM030.fasta      <NA>      water       KI2014\n",
      "101  KI14WSYM031  KI14WSYM031.fasta      <NA>      water       KI2014\n",
      "102  KI14WSYM032  KI14WSYM032.fasta      <NA>      water       KI2014\n",
      "103 KI15aFSYM001 KI15aFSYM001.fasta         1      coral  KI2015a_Pre\n",
      "104 KI15aFSYM003 KI15aFSYM003.fasta         3      coral  KI2015a_Pre\n",
      "105 KI15aFSYM004 KI15aFSYM004.fasta        5b      coral  KI2015a_Pre\n",
      "106 KI15aFSYM005 KI15aFSYM005.fasta     2_572      coral  KI2015a_Pre\n",
      "107 KI15aFSYM006 KI15aFSYM006.fasta        6b      coral  KI2015a_Pre\n",
      "108 KI15aFSYM007 KI15aFSYM007.fasta         7      coral  KI2015a_Pre\n",
      "109 KI15aFSYM008 KI15aFSYM008.fasta        87      coral  KI2015a_Pre\n",
      "110 KI15aSSYM008 KI15aSSYM008.fasta      <NA>   sediment KI2015a_Post\n",
      "111 KI15aFSYM009 KI15aFSYM009.fasta        14      coral  KI2015a_Pre\n",
      "112 KI15aSSYM009 KI15aSSYM009.fasta      <NA>   sediment KI2015a_Post\n",
      "113 KI15aFSYM010 KI15aFSYM010.fasta         9      coral  KI2015a_Pre\n",
      "114 KI15aSSYM010 KI15aSSYM010.fasta      <NA>   sediment KI2015a_Post\n",
      "115 KI15aFSYM011 KI15aFSYM011.fasta       10b      coral  KI2015a_Pre\n",
      "116 KI15aSSYM011 KI15aSSYM011.fasta      <NA>   sediment KI2015a_Post\n",
      "117 KI15aSSYM012 KI15aSSYM012.fasta      <NA>   sediment KI2015a_Post\n",
      "118 KI15aFSYM013 KI15aFSYM013.fasta       574      coral  KI2015a_Pre\n",
      "119 KI15aSSYM013 KI15aSSYM013.fasta      <NA>   sediment KI2015a_Post\n",
      "120 KI15aSSYM014 KI15aSSYM014.fasta      <NA>   sediment KI2015a_Post\n",
      "121 KI15aSSYM015 KI15aSSYM015.fasta      <NA>   sediment KI2015a_Post\n",
      "122 KI15aFSYM016 KI15aFSYM016.fasta        88      coral  KI2015a_Pre\n",
      "123 KI15aSSYM016 KI15aSSYM016.fasta      <NA>   sediment KI2015a_Post\n",
      "124 KI15aFSYM017 KI15aFSYM017.fasta    90_690      coral  KI2015a_Pre\n",
      "125 KI15aSSYM017 KI15aSSYM017.fasta      <NA>   sediment KI2015a_Post\n",
      "126 KI15aSSYM018 KI15aSSYM018.fasta      <NA>   sediment KI2015a_Post\n",
      "127 KI15aFSYM019 KI15aFSYM019.fasta       15b      coral  KI2015a_Pre\n",
      "128 KI15aSSYM019 KI15aSSYM019.fasta      <NA>   sediment KI2015a_Post\n",
      "129 KI15aSSYM020 KI15aSSYM020.fasta      <NA>   sediment KI2015a_Post\n",
      "130 KI15aFSYM021 KI15aFSYM021.fasta    92_860      coral  KI2015a_Pre\n",
      "131 KI15aSSYM021 KI15aSSYM021.fasta      <NA>   sediment KI2015a_Post\n",
      "132 KI15aFSYM022 KI15aFSYM022.fasta        94      coral  KI2015a_Pre\n",
      "133 KI15aSSYM022 KI15aSSYM022.fasta      <NA>   sediment KI2015a_Post\n",
      "134 KI15aSSYM023 KI15aSSYM023.fasta      <NA>   sediment KI2015a_Post\n",
      "135 KI15aSSYM024 KI15aSSYM024.fasta      <NA>   sediment KI2015a_Post\n",
      "136 KI15aSSYM025 KI15aSSYM025.fasta      <NA>   sediment KI2015a_Post\n",
      "137 KI15aSSYM026 KI15aSSYM026.fasta      <NA>   sediment KI2015a_Post\n",
      "138 KI15aSSYM027 KI15aSSYM027.fasta      <NA>   sediment KI2015a_Post\n",
      "139 KI15aSSYM029 KI15aSSYM029.fasta      <NA>   sediment KI2015a_Post\n",
      "140 KI15aSSYM030 KI15aSSYM030.fasta      <NA>   sediment KI2015a_Post\n",
      "141 KI15aFSYM082 KI15aFSYM082.fasta         1      coral KI2015a_Post\n",
      "142 KI15aFSYM083 KI15aFSYM083.fasta     2_572      coral KI2015a_Post\n",
      "143 KI15aFSYM084 KI15aFSYM084.fasta         3      coral KI2015a_Post\n",
      "144 KI15aFSYM085 KI15aFSYM085.fasta         5      coral KI2015a_Post\n",
      "145 KI15aFSYM086 KI15aFSYM086.fasta       632      coral KI2015a_Post\n",
      "146 KI15aFSYM087 KI15aFSYM087.fasta         8      coral KI2015a_Post\n",
      "147 KI15aFSYM089 KI15aFSYM089.fasta        13      coral KI2015a_Post\n",
      "148 KI15aFSYM090 KI15aFSYM090.fasta       634      coral KI2015a_Post\n",
      "149 KI15aFSYM091 KI15aFSYM091.fasta    12_635      coral KI2015a_Post\n",
      "150 KI15aFSYM093 KI15aFSYM093.fasta        15      coral KI2015a_Post\n",
      "151 KI15aFSYM094 KI15aFSYM094.fasta       636      coral KI2015a_Post\n",
      "152 KI15aFSYM095 KI15aFSYM095.fasta        88      coral KI2015a_Post\n",
      "153 KI15aFSYM099 KI15aFSYM099.fasta    92_860      coral KI2015a_Post\n",
      "154 KI15aFSYM100 KI15aFSYM100.fasta        97      coral KI2015a_Post\n",
      "155 KI15aFSYM101 KI15aFSYM101.fasta        94      coral KI2015a_Post\n",
      "156 KI15aFSYM102 KI15aFSYM102.fasta        98      coral KI2015a_Post\n",
      "157 KI15aFSYM106 KI15aFSYM106.fasta       102      coral KI2015a_Post\n",
      "158 KI15aFSYM107 KI15aFSYM107.fasta       106      coral KI2015a_Post\n",
      "159 KI15aFSYM108 KI15aFSYM108.fasta       105      coral KI2015a_Post\n",
      "160 KI15aFSYM109 KI15aFSYM109.fasta       107      coral KI2015a_Post\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "161 KI15aFSYM110 KI15aFSYM110.fasta       109      coral KI2015a_Post\n",
      "162 KI15aFSYM111 KI15aFSYM111.fasta        50      coral KI2015a_Post\n",
      "163 KI15aFSYM114 KI15aFSYM114.fasta        60      coral KI2015a_Post\n",
      "164 KI15aFSYM117 KI15aFSYM117.fasta        63      coral KI2015a_Post\n",
      "165 KI15aFSYM118 KI15aFSYM118.fasta    69_875      coral KI2015a_Post\n",
      "166 KI15aFSYM120 KI15aFSYM120.fasta        76      coral KI2015a_Post\n",
      "167 KI15aFSYM121 KI15aFSYM121.fasta        84      coral KI2015a_Post\n",
      "168 KI15aFSYM122 KI15aFSYM122.fasta      126b      coral KI2015a_Post\n",
      "169 KI15aFSYM126 KI15aFSYM126.fasta       131      coral KI2015a_Post\n",
      "170 KI15aFSYM127 KI15aFSYM127.fasta       129      coral KI2015a_Post\n",
      "171 KI15aFSYM128 KI15aFSYM128.fasta       132      coral KI2015a_Post\n",
      "172 KI15aFSYM129 KI15aFSYM129.fasta       133      coral KI2015a_Post\n",
      "173 KI15aFSYM130 KI15aFSYM130.fasta       134      coral KI2015a_Post\n",
      "174 KI15aFSYM131 KI15aFSYM131.fasta       135      coral KI2015a_Post\n",
      "175 KI15aFSYM132 KI15aFSYM132.fasta       136      coral KI2015a_Post\n",
      "176 KI15aFSYM133 KI15aFSYM133.fasta       137      coral KI2015a_Post\n",
      "177 KI15aFSYM135 KI15aFSYM135.fasta       140      coral KI2015a_Post\n",
      "178 KI15aFSYM136 KI15aFSYM136.fasta       142      coral KI2015a_Post\n",
      "179 KI15aFSYM137 KI15aFSYM137.fasta       144      coral KI2015a_Post\n",
      "180 KI15aFSYM138 KI15aFSYM138.fasta       143      coral KI2015a_Post\n",
      "181 KI15aFSYM140 KI15aFSYM140.fasta       149      coral KI2015a_Post\n",
      "182 KI15aFSYM141 KI15aFSYM141.fasta       147      coral KI2015a_Post\n",
      "183 KI15aFSYM143 KI15aFSYM143.fasta       151      coral KI2015a_Post\n",
      "184 KI15aFSYM144 KI15aFSYM144.fasta       385      coral KI2015a_Post\n",
      "185 KI15aFSYM146 KI15aFSYM146.fasta       387      coral KI2015a_Post\n",
      "186 KI15aFSYM147 KI15aFSYM147.fasta       474      coral KI2015a_Post\n",
      "187 KI15aFSYM148 KI15aFSYM148.fasta       476      coral KI2015a_Post\n",
      "188 KI15aFSYM149 KI15aFSYM149.fasta       477      coral KI2015a_Post\n",
      "189 KI15aFSYM152 KI15aFSYM152.fasta       469      coral KI2015a_Post\n",
      "190 KI15aFSYM154 KI15aFSYM154.fasta       468      coral KI2015a_Post\n",
      "191 KI15aFSYM155 KI15aFSYM155.fasta       467      coral KI2015a_Post\n",
      "192 KI15aFSYM157 KI15aFSYM157.fasta       448      coral KI2015a_Post\n",
      "193 KI15aFSYM158 KI15aFSYM158.fasta   451_871      coral KI2015a_Post\n",
      "194 KI15aFSYM159 KI15aFSYM159.fasta       456      coral KI2015a_Post\n",
      "195 KI15aFSYM161 KI15aFSYM161.fasta       453      coral KI2015a_Post\n",
      "196 KI15aFSYM162 KI15aFSYM162.fasta       265      coral KI2015a_Post\n",
      "197 KI15aFSYM165 KI15aFSYM165.fasta        41      coral KI2015a_Post\n",
      "198 KI15aFSYM166 KI15aFSYM166.fasta        43      coral KI2015a_Post\n",
      "199 KI15aFSYM167 KI15aFSYM167.fasta        44      coral KI2015a_Post\n",
      "200 KI15aWSYM005 KI15aWSYM005.fasta      <NA>      water KI2015a_Post\n",
      "201 KI15aWSYM006 KI15aWSYM006.fasta      <NA>      water KI2015a_Post\n",
      "202 KI15aWSYM007 KI15aWSYM007.fasta      <NA>      water KI2015a_Post\n",
      "203 KI15aWSYM008 KI15aWSYM008.fasta      <NA>      water KI2015a_Post\n",
      "204 KI15aWSYM009 KI15aWSYM009.fasta      <NA>      water KI2015a_Post\n",
      "205 KI15aWSYM010 KI15aWSYM010.fasta      <NA>      water KI2015a_Post\n",
      "206 KI15aWSYM011 KI15aWSYM011.fasta      <NA>      water KI2015a_Post\n",
      "207 KI15aWSYM012 KI15aWSYM012.fasta      <NA>      water KI2015a_Post\n",
      "208 KI15aWSYM013 KI15aWSYM013.fasta      <NA>      water KI2015a_Post\n",
      "209 KI15aWSYM014 KI15aWSYM014.fasta      <NA>      water KI2015a_Post\n",
      "210 KI15aWSYM015 KI15aWSYM015.fasta      <NA>      water KI2015a_Post\n",
      "211 KI15aWSYM016 KI15aWSYM016.fasta      <NA>      water KI2015a_Post\n",
      "212 KI15aWSYM017 KI15aWSYM017.fasta      <NA>      water KI2015a_Post\n",
      "213 KI15aWSYM018 KI15aWSYM018.fasta      <NA>      water KI2015a_Post\n",
      "214 KI15aWSYM019 KI15aWSYM019.fasta      <NA>      water KI2015a_Post\n",
      "215 KI15aWSYM020 KI15aWSYM020.fasta      <NA>      water KI2015a_Post\n",
      "216 KI15bFSYM001 KI15bFSYM001.fasta         1      coral      KI2015b\n",
      "217 KI15bSSYM001 KI15bSSYM001.fasta      <NA>   sediment      KI2015b\n",
      "218 KI15bSSYM002 KI15bSSYM002.fasta      <NA>   sediment      KI2015b\n",
      "219 KI15bFSYM003 KI15bFSYM003.fasta         3      coral      KI2015b\n",
      "220 KI15bSSYM003 KI15bSSYM003.fasta      <NA>   sediment      KI2015b\n",
      "221 KI15bFSYM004 KI15bFSYM004.fasta     2_572      coral      KI2015b\n",
      "222 KI15bSSYM004 KI15bSSYM004.fasta      <NA>   sediment      KI2015b\n",
      "223 KI15bFSYM005 KI15bFSYM005.fasta         5      coral      KI2015b\n",
      "224 KI15bSSYM005 KI15bSSYM005.fasta      <NA>   sediment      KI2015b\n",
      "225 KI15bFSYM006 KI15bFSYM006.fasta         8      coral      KI2015b\n",
      "226 KI15bSSYM006 KI15bSSYM006.fasta      <NA>   sediment      KI2015b\n",
      "227 KI15bSSYM007 KI15bSSYM007.fasta      <NA>   sediment      KI2015b\n",
      "228 KI15bSSYM008 KI15bSSYM008.fasta      <NA>   sediment      KI2015b\n",
      "229 KI15bFSYM009 KI15bFSYM009.fasta       634      coral      KI2015b\n",
      "230 KI15bSSYM009 KI15bSSYM009.fasta      <NA>   sediment      KI2015b\n",
      "231 KI15bFSYM010 KI15bFSYM010.fasta    12_635      coral      KI2015b\n",
      "232 KI15bSSYM010 KI15bSSYM010.fasta      <NA>   sediment      KI2015b\n",
      "233 KI15bFSYM012 KI15bFSYM012.fasta       574      coral      KI2015b\n",
      "234 KI15bSSYM012 KI15bSSYM012.fasta      <NA>   sediment      KI2015b\n",
      "235 KI15bFSYM013 KI15bFSYM013.fasta        87      coral      KI2015b\n",
      "236 KI15bSSYM013 KI15bSSYM013.fasta      <NA>   sediment      KI2015b\n",
      "237 KI15bFSYM014 KI15bFSYM014.fasta        13      coral      KI2015b\n",
      "238 KI15bSSYM014 KI15bSSYM014.fasta      <NA>   sediment      KI2015b\n",
      "239 KI15bSSYM015 KI15bSSYM015.fasta      <NA>   sediment      KI2015b\n",
      "240 KI15bFSYM016 KI15bFSYM016.fasta       636      coral      KI2015b\n",
      "241 KI15bSSYM016 KI15bSSYM016.fasta      <NA>   sediment      KI2015b\n",
      "242 KI15bFSYM017 KI15bFSYM017.fasta        15      coral      KI2015b\n",
      "243 KI15bSSYM017 KI15bSSYM017.fasta      <NA>   sediment      KI2015b\n",
      "244 KI15bFSYM018 KI15bFSYM018.fasta        88      coral      KI2015b\n",
      "245 KI15bSSYM018 KI15bSSYM018.fasta      <NA>   sediment      KI2015b\n",
      "246 KI15bSSYM019 KI15bSSYM019.fasta      <NA>   sediment      KI2015b\n",
      "247 KI15bFSYM021 KI15bFSYM021.fasta    92_860      coral      KI2015b\n",
      "248 KI15bSSYM021 KI15bSSYM021.fasta      <NA>   sediment      KI2015b\n",
      "249 KI15bSSYM022 KI15bSSYM022.fasta      <NA>   sediment      KI2015b\n",
      "250 KI15bSSYM023 KI15bSSYM023.fasta      <NA>   sediment      KI2015b\n",
      "251 KI15bFSYM024 KI15bFSYM024.fasta        97      coral      KI2015b\n",
      "252 KI15bSSYM024 KI15bSSYM024.fasta      <NA>   sediment      KI2015b\n",
      "253 KI15bFSYM025 KI15bFSYM025.fasta        98      coral      KI2015b\n",
      "254 KI15bFSYM027 KI15bFSYM027.fasta       102      coral      KI2015b\n",
      "255 KI15bFSYM028 KI15bFSYM028.fasta       106      coral      KI2015b\n",
      "256 KI15bFSYM029 KI15bFSYM029.fasta       105      coral      KI2015b\n",
      "257 KI15bFSYM030 KI15bFSYM030.fasta       110      coral      KI2015b\n",
      "258 KI15bFSYM034 KI15bFSYM034.fasta       116      coral      KI2015b\n",
      "259 KI15bFSYM035 KI15bFSYM035.fasta       119      coral      KI2015b\n",
      "260 KI15bFSYM036 KI15bFSYM036.fasta       120      coral      KI2015b\n",
      "261 KI15bFSYM037 KI15bFSYM037.fasta       132      coral      KI2015b\n",
      "262 KI15bFSYM040 KI15bFSYM040.fasta       129      coral      KI2015b\n",
      "263 KI15bFSYM041 KI15bFSYM041.fasta       131      coral      KI2015b\n",
      "264 KI15bFSYM043 KI15bFSYM043.fasta       134      coral      KI2015b\n",
      "265 KI15bFSYM044 KI15bFSYM044.fasta       135      coral      KI2015b\n",
      "266 KI15bFSYM045 KI15bFSYM045.fasta       136      coral      KI2015b\n",
      "267 KI15bFSYM046 KI15bFSYM046.fasta       137      coral      KI2015b\n",
      "268 KI15bFSYM047 KI15bFSYM047.fasta       139      coral      KI2015b\n",
      "269 KI15bFSYM049 KI15bFSYM049.fasta       144      coral      KI2015b\n",
      "270 KI15bFSYM051 KI15bFSYM051.fasta       145      coral      KI2015b\n",
      "271 KI15bFSYM052 KI15bFSYM052.fasta       149      coral      KI2015b\n",
      "272 KI15bFSYM054 KI15bFSYM054.fasta       151      coral      KI2015b\n",
      "273 KI15bFSYM055 KI15bFSYM055.fasta       152      coral      KI2015b\n",
      "274 KI15bFSYM056 KI15bFSYM056.fasta       385      coral      KI2015b\n",
      "275 KI15bFSYM058 KI15bFSYM058.fasta       387      coral      KI2015b\n",
      "276 KI15bFSYM059 KI15bFSYM059.fasta       428      coral      KI2015b\n",
      "277 KI15bFSYM060 KI15bFSYM060.fasta       427      coral      KI2015b\n",
      "278 KI15bFSYM062 KI15bFSYM062.fasta       430      coral      KI2015b\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "279 KI15bFSYM063 KI15bFSYM063.fasta       434      coral      KI2015b\n",
      "280 KI15bFSYM064 KI15bFSYM064.fasta       436      coral      KI2015b\n",
      "281 KI15bFSYM066 KI15bFSYM066.fasta       437      coral      KI2015b\n",
      "282 KI15bFSYM070 KI15bFSYM070.fasta       443      coral      KI2015b\n",
      "283 KI15bFSYM071 KI15bFSYM071.fasta       440      coral      KI2015b\n",
      "284 KI15bFSYM073 KI15bFSYM073.fasta       446      coral      KI2015b\n",
      "285 KI15bFSYM074 KI15bFSYM074.fasta       441      coral      KI2015b\n",
      "286 KI15bFSYM128 KI15bFSYM128.fasta       682      coral      KI2015b\n",
      "287 KI15bFSYM129 KI15bFSYM129.fasta       133      coral      KI2015b\n",
      "288 KI15bFSYM130 KI15bFSYM130.fasta       147      coral      KI2015b\n",
      "289 KI15bFSYM133 KI15bFSYM133.fasta   445_683      coral      KI2015b\n",
      "290 KI15bFSYM134 KI15bFSYM134.fasta       476      coral      KI2015b\n",
      "291 KI15bFSYM135 KI15bFSYM135.fasta       474      coral      KI2015b\n",
      "292 KI15bFSYM137 KI15bFSYM137.fasta       477      coral      KI2015b\n",
      "293 KI15bFSYM141 KI15bFSYM141.fasta   451_871      coral      KI2015b\n",
      "294 KI15bFSYM142 KI15bFSYM142.fasta       456      coral      KI2015b\n",
      "295 KI15bFSYM143 KI15bFSYM143.fasta       469      coral      KI2015b\n",
      "296 KI15bFSYM144 KI15bFSYM144.fasta        44      coral      KI2015b\n",
      "297 KI15bFSYM145 KI15bFSYM145.fasta        46      coral      KI2015b\n",
      "298 KI15bFSYM146 KI15bFSYM146.fasta       684      coral      KI2015b\n",
      "299 KI15bFSYM150 KI15bFSYM150.fasta       688      coral      KI2015b\n",
      "300 KI15bFSYM152 KI15bFSYM152.fasta    90_690      coral      KI2015b\n",
      "301 KI15bFSYM154 KI15bFSYM154.fasta       109      coral      KI2015b\n",
      "302 KI15bFSYM155 KI15bFSYM155.fasta   114_692      coral      KI2015b\n",
      "303 KI15bFSYM158 KI15bFSYM158.fasta   121_693      coral      KI2015b\n",
      "304 KI15bFSYM159 KI15bFSYM159.fasta      122b      coral      KI2015b\n",
      "305 KI15bFSYM160 KI15bFSYM160.fasta       123      coral      KI2015b\n",
      "306 KI15bFSYM161 KI15bFSYM161.fasta   124_694      coral      KI2015b\n",
      "307 KI15bFSYM165 KI15bFSYM165.fasta        60      coral      KI2015b\n",
      "308 KI15bFSYM168 KI15bFSYM168.fasta        63      coral      KI2015b\n",
      "309 KI15bFSYM170 KI15bFSYM170.fasta    69_875      coral      KI2015b\n",
      "310 KI15bFSYM172 KI15bFSYM172.fasta        76      coral      KI2015b\n",
      "311 KI15bFSYM173 KI15bFSYM173.fasta        84      coral      KI2015b\n",
      "312 KI15bFSYM174 KI15bFSYM174.fasta        85      coral      KI2015b\n",
      "313 KI15bFSYM175 KI15bFSYM175.fasta        86      coral      KI2015b\n",
      "314 KI15bFSYM176 KI15bFSYM176.fasta       241      coral      KI2015b\n",
      "315 KI15bFSYM180 KI15bFSYM180.fasta   256_697      coral      KI2015b\n",
      "316 KI15bFSYM181 KI15bFSYM181.fasta       698      coral      KI2015b\n",
      "317 KI15bFSYM182 KI15bFSYM182.fasta       717      coral      KI2015b\n",
      "318 KI15bFSYM183 KI15bFSYM183.fasta       718      coral      KI2015b\n",
      "319 KI15bFSYM199 KI15bFSYM199.fasta       721      coral      KI2015b\n",
      "320 KI15bFSYM201 KI15bFSYM201.fasta       723      coral      KI2015b\n",
      "321 KI15bFSYM208 KI15bFSYM208.fasta       730      coral      KI2015b\n",
      "322 KI15bFSYM209 KI15bFSYM209.fasta       731      coral      KI2015b\n",
      "323 KI15bFSYM210 KI15bFSYM210.fasta       732      coral      KI2015b\n",
      "324 KI15bFSYM211 KI15bFSYM211.fasta       733      coral      KI2015b\n",
      "325 KI15bFSYM250 KI15bFSYM250.fasta  314_1088      coral      KI2015b\n",
      "326 KI15bFSYM251 KI15bFSYM251.fasta   317_949      coral      KI2015b\n",
      "327 KI15bFSYM252 KI15bFSYM252.fasta       323      coral      KI2015b\n",
      "328 KI15bFSYM253 KI15bFSYM253.fasta       324      coral      KI2015b\n",
      "329 KI15bFSYM254 KI15bFSYM254.fasta       325      coral      KI2015b\n",
      "330 KI15bFSYM256 KI15bFSYM256.fasta       329      coral      KI2015b\n",
      "331 KI15bFSYM257 KI15bFSYM257.fasta       331      coral      KI2015b\n",
      "332 KI15bFSYM258 KI15bFSYM258.fasta       336      coral      KI2015b\n",
      "333 KI15bFSYM261 KI15bFSYM261.fasta       348      coral      KI2015b\n",
      "334 KI15bFSYM262 KI15bFSYM262.fasta       349      coral      KI2015b\n",
      "335 KI15bFSYM264 KI15bFSYM264.fasta       354      coral      KI2015b\n",
      "336 KI15bFSYM265 KI15bFSYM265.fasta       355      coral      KI2015b\n",
      "337 KI15bFSYM266 KI15bFSYM266.fasta       363      coral      KI2015b\n",
      "338 KI15bFSYM267 KI15bFSYM267.fasta       361      coral      KI2015b\n",
      "339 KI15bFSYM268 KI15bFSYM268.fasta       359      coral      KI2015b\n",
      "340 KI15bFSYM269 KI15bFSYM269.fasta       773      coral      KI2015b\n",
      "341 KI15bFSYM270 KI15bFSYM270.fasta       774      coral      KI2015b\n",
      "342 KI15bFSYM271 KI15bFSYM271.fasta       775      coral      KI2015b\n",
      "343 KI15bFSYM272 KI15bFSYM272.fasta       776      coral      KI2015b\n",
      "344 KI15bFSYM273 KI15bFSYM273.fasta       777      coral      KI2015b\n",
      "345 KI15bFSYM274 KI15bFSYM274.fasta       778      coral      KI2015b\n",
      "346 KI15bFSYM275 KI15bFSYM275.fasta       779      coral      KI2015b\n",
      "347 KI15bFSYM276 KI15bFSYM276.fasta       780      coral      KI2015b\n",
      "348 KI15bFSYM278 KI15bFSYM278.fasta       782      coral      KI2015b\n",
      "349 KI15bFSYM280 KI15bFSYM280.fasta       784      coral      KI2015b\n",
      "350 KI15bFSYM281 KI15bFSYM281.fasta       785      coral      KI2015b\n",
      "351 KI15bFSYM337 KI15bFSYM337.fasta       841      coral      KI2015b\n",
      "352 KI15bFSYM338 KI15bFSYM338.fasta       842      coral      KI2015b\n",
      "353 KI15bFSYM339 KI15bFSYM339.fasta       843      coral      KI2015b\n",
      "354 KI15bFSYM341 KI15bFSYM341.fasta       845      coral      KI2015b\n",
      "355 KI15bFSYM342 KI15bFSYM342.fasta       846      coral      KI2015b\n",
      "356 KI15bFSYM343 KI15bFSYM343.fasta       847      coral      KI2015b\n",
      "357 KI15bFSYM344 KI15bFSYM344.fasta       848      coral      KI2015b\n",
      "358 KI15bFSYM347 KI15bFSYM347.fasta       851      coral      KI2015b\n",
      "359 KI15bFSYM350 KI15bFSYM350.fasta       854      coral      KI2015b\n",
      "360 KI15bWSYM001 KI15bWSYM001.fasta      <NA>      water      KI2015b\n",
      "361 KI15bWSYM002 KI15bWSYM002.fasta      <NA>      water      KI2015b\n",
      "362 KI15bWSYM003 KI15bWSYM003.fasta      <NA>      water      KI2015b\n",
      "363 KI15bWSYM004 KI15bWSYM004.fasta      <NA>      water      KI2015b\n",
      "364 KI15bWSYM005 KI15bWSYM005.fasta      <NA>      water      KI2015b\n",
      "365 KI15bWSYM006 KI15bWSYM006.fasta      <NA>      water      KI2015b\n",
      "366 KI15bWSYM007 KI15bWSYM007.fasta      <NA>      water      KI2015b\n",
      "367 KI15bWSYM008 KI15bWSYM008.fasta      <NA>      water      KI2015b\n",
      "368 KI15bWSYM017 KI15bWSYM017.fasta      <NA>      water      KI2015b\n",
      "369 KI15bWSYM018 KI15bWSYM018.fasta      <NA>      water      KI2015b\n",
      "370 KI15bWSYM019 KI15bWSYM019.fasta      <NA>      water      KI2015b\n",
      "371 KI15bWSYM020 KI15bWSYM020.fasta      <NA>      water      KI2015b\n",
      "372 KI15bWSYM023 KI15bWSYM023.fasta      <NA>      water      KI2015b\n",
      "373 KI15bWSYM024 KI15bWSYM024.fasta      <NA>      water      KI2015b\n",
      "374 KI15cSSYM007 KI15cSSYM007.fasta      <NA>   sediment      KI2015c\n",
      "375 KI15cSSYM008 KI15cSSYM008.fasta      <NA>   sediment      KI2015c\n",
      "376 KI15cSSYM009 KI15cSSYM009.fasta      <NA>   sediment      KI2015c\n",
      "377 KI15cSSYM039 KI15cSSYM039.fasta      <NA>   sediment      KI2015c\n",
      "378 KI15cSSYM040 KI15cSSYM040.fasta      <NA>   sediment      KI2015c\n",
      "379 KI15cSSYM042 KI15cSSYM042.fasta      <NA>   sediment      KI2015c\n",
      "380 KI15cSSYM051 KI15cSSYM051.fasta      <NA>   sediment      KI2015c\n",
      "381 KI15cSSYM052 KI15cSSYM052.fasta      <NA>   sediment      KI2015c\n",
      "382 KI15cSSYM054 KI15cSSYM054.fasta      <NA>   sediment      KI2015c\n",
      "    TubeNumber         Coral_Species site Status Year_Pre_Post\n",
      "1            1        Porites_lobata   35  alive          2014\n",
      "2            2     Montipora_foliosa   35   gone          2014\n",
      "3            3   Pocillopora_eydouxi   35     UK          2014\n",
      "4            3              sediment    8   <NA>          2014\n",
      "5            4              sediment    8   <NA>          2014\n",
      "6            5        Porites_lobata   35  alive          2014\n",
      "7            5              sediment    8   <NA>          2014\n",
      "8            6              sediment    8   <NA>          2014\n",
      "9            7     Montipora_foliosa   35     UK          2014\n",
      "10           7              sediment    8   <NA>          2014\n",
      "11           8        Porites_lobata   35     UK          2014\n",
      "12           8              sediment    8   <NA>          2014\n",
      "13           9   Pocillopora_eydouxi   35     UK          2014\n",
      "14          10              sediment   35   <NA>          2014\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15          11              sediment   35   <NA>          2014\n",
      "16          12        Porites_lobata   35   gone          2014\n",
      "17          12              sediment   35   <NA>          2014\n",
      "18          13   Pocillopora_eydouxi   35   dead          2014\n",
      "19          13              sediment   35   <NA>          2014\n",
      "20          14              sediment   35   <NA>          2014\n",
      "21          15        Porites_lobata   35  alive          2014\n",
      "22          15              sediment   30   <NA>          2014\n",
      "23          16              sediment   30   <NA>          2014\n",
      "24          17              sediment   30   <NA>          2014\n",
      "25          18              sediment   30   <NA>          2014\n",
      "26          19              sediment   30   <NA>          2014\n",
      "27          20              sediment   30   <NA>          2014\n",
      "28          41     Montipora_foliosa   27     UK          2014\n",
      "29          43   Pocillopora_eydouxi   27     UK          2014\n",
      "30          44     Montipora_foliosa   27   gone          2014\n",
      "31          50   Pocillopora_eydouxi    8     UK          2014\n",
      "32          50              sediment   35   <NA>          2014\n",
      "33          58              sediment   27   <NA>          2014\n",
      "34          59              sediment   27   <NA>          2014\n",
      "35          60     Montipora_foliosa    8   dead          2014\n",
      "36          60              sediment   27   <NA>          2014\n",
      "37          61              sediment   27   <NA>          2014\n",
      "38          63        Porites_lobata    8  alive          2014\n",
      "39          63              sediment   27   <NA>          2014\n",
      "40          69     Montipora_foliosa    8     UK          2014\n",
      "41          76     Montipora_foliosa    8   dead          2014\n",
      "42          84   Pocillopora_eydouxi    8   dead          2014\n",
      "43          87        Porites_lobata   35     UK          2014\n",
      "44          88   Pocillopora_eydouxi   35   dead          2014\n",
      "45          90     Montipora_foliosa   35   dead          2014\n",
      "46          92   Pocillopora_eydouxi   35   gone          2014\n",
      "47          94     Montipora_foliosa   35   dead          2014\n",
      "48          97        Porites_lobata   35     UK          2014\n",
      "49          98     Montipora_foliosa   35   dead          2014\n",
      "50         102     Montipora_foliosa   35   dead          2014\n",
      "51         105     Montipora_foliosa   35   dead          2014\n",
      "52         106   Pocillopora_eydouxi   35   dead          2014\n",
      "53         107        Porites_lobata   35     UK          2014\n",
      "54         109   Pocillopora_eydouxi   35   dead          2014\n",
      "55         126        Porites_lobata   30     UK          2014\n",
      "56         129   Pocillopora_eydouxi   30  alive          2014\n",
      "57         131        Porites_lobata   30     UK          2014\n",
      "58         132   Pocillopora_eydouxi   30   dead          2014\n",
      "59         133     Montipora_foliosa   30   gone          2014\n",
      "60         134     Montipora_foliosa   30   gone          2014\n",
      "61         135   Pocillopora_eydouxi   30   gone          2014\n",
      "62         136        Porites_lobata   30   gone          2014\n",
      "63         137     Montipora_foliosa   30   gone          2014\n",
      "64         140        Porites_lobata   30     UK          2014\n",
      "65         142   Pocillopora_eydouxi   30     UK          2014\n",
      "66         143     Montipora_foliosa   30   dead          2014\n",
      "67         144        Porites_lobata   30   gone          2014\n",
      "68         147     Montipora_foliosa   30     UK          2014\n",
      "69         149   Pocillopora_eydouxi   30   dead          2014\n",
      "70         151        Porites_lobata   30  alive          2014\n",
      "71         265     Montipora_foliosa   27     UK          2014\n",
      "72         385     Montipora_foliosa   30   dead          2014\n",
      "73         387        Porites_lobata   30   gone          2014\n",
      "74         448     Montipora_foliosa   27     UK          2014\n",
      "75         451        Porites_lobata   27  alive          2014\n",
      "76         453        Porites_lobata   27     UK          2014\n",
      "77         456     Montipora_foliosa   27     UK          2014\n",
      "78         467        Porites_lobata   27     UK          2014\n",
      "79         468   Pocillopora_eydouxi   27     UK          2014\n",
      "80         469   Pocillopora_eydouxi   27     UK          2014\n",
      "81         474        Porites_lobata   27   gone          2014\n",
      "82         476        Porites_lobata   27   gone          2014\n",
      "83           1                 water   35   <NA>          2014\n",
      "84           3                 water    8   <NA>          2014\n",
      "85           4                 water    8   <NA>          2014\n",
      "86           5                 water   35   <NA>          2014\n",
      "87           6                 water   35   <NA>          2014\n",
      "88           7                 water   35   <NA>          2014\n",
      "89           8                 water   30   <NA>          2014\n",
      "90          11                 water   27   <NA>          2014\n",
      "91          12                 water   27   <NA>          2014\n",
      "92          13                 water   27   <NA>          2014\n",
      "93          14                 water   27   <NA>          2014\n",
      "94          15                 water   30   <NA>          2014\n",
      "95          16                 water   30   <NA>          2014\n",
      "96          17                 water   30   <NA>          2014\n",
      "97          21                 water    8   <NA>          2014\n",
      "98          28                 water   35   <NA>          2014\n",
      "99          29                 water   35   <NA>          2014\n",
      "100         30                 water   35   <NA>          2014\n",
      "101         31                 water    8   <NA>          2014\n",
      "102         32                 water   30   <NA>          2014\n",
      "103          1        Porites_lobata   35  alive   2015Jan_Pre\n",
      "104          3   Pocillopora_eydouxi   35     UK   2015Jan_Pre\n",
      "105          4   Pocillopora_eydouxi   35   <NA>   2015Jan_Pre\n",
      "106          5     Montipora_foliosa   35   gone   2015Jan_Pre\n",
      "107          6        Porites_lobata   35   <NA>   2015Jan_Pre\n",
      "108          7     Montipora_foliosa   35     UK   2015Jan_Pre\n",
      "109          8        Porites_lobata   35     UK   2015Jan_Pre\n",
      "110          8              sediment   35   <NA>  2015Jan_Post\n",
      "111          9     Montipora_foliosa   35   <NA>   2015Jan_Pre\n",
      "112          9              sediment   35   <NA>  2015Jan_Post\n",
      "113         10   Pocillopora_eydouxi   35     UK   2015Jan_Pre\n",
      "114         10              sediment   35   <NA>  2015Jan_Post\n",
      "115         11        Porites_lobata   35   <NA>   2015Jan_Pre\n",
      "116         11              sediment   35   <NA>  2015Jan_Post\n",
      "117         12              sediment   35   <NA>  2015Jan_Post\n",
      "118         13     Montipora_foliosa   35   gone   2015Jan_Pre\n",
      "119         13              sediment   30   <NA>  2015Jan_Post\n",
      "120         14              sediment   30   <NA>  2015Jan_Post\n",
      "121         15              sediment   30   <NA>  2015Jan_Post\n",
      "122         16   Pocillopora_eydouxi   35   dead   2015Jan_Pre\n",
      "123         16              sediment   30   <NA>  2015Jan_Post\n",
      "124         17     Montipora_foliosa   35   dead   2015Jan_Pre\n",
      "125         17              sediment   30   <NA>  2015Jan_Post\n",
      "126         18              sediment   30   <NA>  2015Jan_Post\n",
      "127         19   Pocillopora_eydouxi   35   <NA>   2015Jan_Pre\n",
      "128         19              sediment   27   <NA>  2015Jan_Post\n",
      "129         20              sediment   27   <NA>  2015Jan_Post\n",
      "130         21   Pocillopora_eydouxi   35   gone   2015Jan_Pre\n",
      "131         21              sediment   27   <NA>  2015Jan_Post\n",
      "132         22     Montipora_foliosa   35   dead   2015Jan_Pre\n",
      "133         22              sediment   27   <NA>  2015Jan_Post\n",
      "134         23              sediment   27   <NA>  2015Jan_Post\n",
      "135         24              sediment   27   <NA>  2015Jan_Post\n",
      "136         25              sediment    8   <NA>  2015Jan_Post\n",
      "137         26              sediment    8   <NA>  2015Jan_Post\n",
      "138         27              sediment    8   <NA>  2015Jan_Post\n",
      "139         29              sediment    8   <NA>  2015Jan_Post\n",
      "140         30              sediment    8   <NA>  2015Jan_Post\n",
      "141         82        Porites_lobata   35  alive  2015Jan_Post\n",
      "142         83     Montipora_foliosa   35   gone  2015Jan_Post\n",
      "143         84   Pocillopora_eydouxi   35     UK  2015Jan_Post\n",
      "144         85        Porites_lobata   35  alive  2015Jan_Post\n",
      "145         86   Pocillopora_eydouxi   35   dead  2015Jan_Post\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "146         87        Porites_lobata   35     UK  2015Jan_Post\n",
      "147         89   Pocillopora_eydouxi   35   dead  2015Jan_Post\n",
      "148         90     Montipora_foliosa   35   dead  2015Jan_Post\n",
      "149         91        Porites_lobata   35   gone  2015Jan_Post\n",
      "150         93        Porites_lobata   35  alive  2015Jan_Post\n",
      "151         94   Pocillopora_eydouxi   35     UK  2015Jan_Post\n",
      "152         95   Pocillopora_eydouxi   35   dead  2015Jan_Post\n",
      "153         99   Pocillopora_eydouxi   35   gone  2015Jan_Post\n",
      "154        100        Porites_lobata   35     UK  2015Jan_Post\n",
      "155        101     Montipora_foliosa   35   dead  2015Jan_Post\n",
      "156        102     Montipora_foliosa   35   dead  2015Jan_Post\n",
      "157        106     Montipora_foliosa   35   dead  2015Jan_Post\n",
      "158        107   Pocillopora_eydouxi   35   dead  2015Jan_Post\n",
      "159        108     Montipora_foliosa   35   dead  2015Jan_Post\n",
      "160        109        Porites_lobata   35     UK  2015Jan_Post\n",
      "161        110   Pocillopora_eydouxi   35   dead  2015Jan_Post\n",
      "162        111   Pocillopora_eydouxi    8     UK  2015Jan_Post\n",
      "163        114     Montipora_foliosa    8   dead  2015Jan_Post\n",
      "164        117        Porites_lobata    8  alive  2015Jan_Post\n",
      "165        118     Montipora_foliosa    8     UK  2015Jan_Post\n",
      "166        120     Montipora_foliosa    8   dead  2015Jan_Post\n",
      "167        121   Pocillopora_eydouxi    8   dead  2015Jan_Post\n",
      "168        122        Porites_lobata   30     UK  2015Jan_Post\n",
      "169        126        Porites_lobata   30     UK  2015Jan_Post\n",
      "170        127   Pocillopora_eydouxi   30  alive  2015Jan_Post\n",
      "171        128   Pocillopora_eydouxi   30   dead  2015Jan_Post\n",
      "172        129     Montipora_foliosa   30   gone  2015Jan_Post\n",
      "173        130     Montipora_foliosa   30   gone  2015Jan_Post\n",
      "174        131   Pocillopora_eydouxi   30   gone  2015Jan_Post\n",
      "175        132        Porites_lobata   30   gone  2015Jan_Post\n",
      "176        133     Montipora_foliosa   30   gone  2015Jan_Post\n",
      "177        135        Porites_lobata   30     UK  2015Jan_Post\n",
      "178        136   Pocillopora_eydouxi   30     UK  2015Jan_Post\n",
      "179        137        Porites_lobata   30   gone  2015Jan_Post\n",
      "180        138     Montipora_foliosa   30   dead  2015Jan_Post\n",
      "181        140   Pocillopora_eydouxi   30   dead  2015Jan_Post\n",
      "182        141     Montipora_foliosa   30     UK  2015Jan_Post\n",
      "183        143        Porites_lobata   30  alive  2015Jan_Post\n",
      "184        144     Montipora_foliosa   30   dead  2015Jan_Post\n",
      "185        146        Porites_lobata   30   gone  2015Jan_Post\n",
      "186        147        Porites_lobata   27   gone  2015Jan_Post\n",
      "187        148        Porites_lobata   27   gone  2015Jan_Post\n",
      "188        149   Pocillopora_eydouxi   27   dead  2015Jan_Post\n",
      "189        152   Pocillopora_eydouxi   27     UK  2015Jan_Post\n",
      "190        154   Pocillopora_eydouxi   27     UK  2015Jan_Post\n",
      "191        155        Porites_lobata   27     UK  2015Jan_Post\n",
      "192        157     Montipora_foliosa   27     UK  2015Jan_Post\n",
      "193        158        Porites_lobata   27  alive  2015Jan_Post\n",
      "194        159     Montipora_foliosa   27     UK  2015Jan_Post\n",
      "195        161        Porites_lobata   27     UK  2015Jan_Post\n",
      "196        162     Montipora_foliosa   27     UK  2015Jan_Post\n",
      "197        165     Montipora_foliosa   27     UK  2015Jan_Post\n",
      "198        166   Pocillopora_eydouxi   27     UK  2015Jan_Post\n",
      "199        167     Montipora_foliosa   27   gone  2015Jan_Post\n",
      "200          5                 water    8   <NA>  2015Jan_Post\n",
      "201          6                 water    8   <NA>  2015Jan_Post\n",
      "202          7                 water    8   <NA>  2015Jan_Post\n",
      "203          8                 water    8   <NA>  2015Jan_Post\n",
      "204          9                 water   35   <NA>  2015Jan_Post\n",
      "205         10                 water   35   <NA>  2015Jan_Post\n",
      "206         11                 water   35   <NA>  2015Jan_Post\n",
      "207         12                 water   35   <NA>  2015Jan_Post\n",
      "208         13                 water   27   <NA>  2015Jan_Post\n",
      "209         14                 water   27   <NA>  2015Jan_Post\n",
      "210         15                 water   27   <NA>  2015Jan_Post\n",
      "211         16                 water   27   <NA>  2015Jan_Post\n",
      "212         17                 water   30   <NA>  2015Jan_Post\n",
      "213         18                 water   30   <NA>  2015Jan_Post\n",
      "214         19                 water   30   <NA>  2015Jan_Post\n",
      "215         20                 water   30   <NA>  2015Jan_Post\n",
      "216          1        Porites_lobata   35  alive       2015May\n",
      "217          1              sediment   35   <NA>       2015May\n",
      "218          2              sediment   35   <NA>       2015May\n",
      "219          3   Pocillopora_eydouxi   35     UK       2015May\n",
      "220          3              sediment   35   <NA>       2015May\n",
      "221          4     Montipora_foliosa   35   gone       2015May\n",
      "222          4              sediment   35   <NA>       2015May\n",
      "223          5        Porites_lobata   35  alive       2015May\n",
      "224          5              sediment   35   <NA>       2015May\n",
      "225          6        Porites_lobata   35     UK       2015May\n",
      "226          6              sediment   35   <NA>       2015May\n",
      "227          7              sediment   30   <NA>       2015May\n",
      "228          8              sediment   30   <NA>       2015May\n",
      "229          9     Montipora_foliosa   35   dead       2015May\n",
      "230          9              sediment   30   <NA>       2015May\n",
      "231         10        Porites_lobata   35   gone       2015May\n",
      "232         10              sediment   30   <NA>       2015May\n",
      "233         12     Montipora_foliosa   35   gone       2015May\n",
      "234         12              sediment   30   <NA>       2015May\n",
      "235         13        Porites_lobata   35     UK       2015May\n",
      "236         13              sediment   27   <NA>       2015May\n",
      "237         14   Pocillopora_eydouxi   35   dead       2015May\n",
      "238         14              sediment   27   <NA>       2015May\n",
      "239         15              sediment   27   <NA>       2015May\n",
      "240         16   Pocillopora_eydouxi   35     UK       2015May\n",
      "241         16              sediment   27   <NA>       2015May\n",
      "242         17        Porites_lobata   35  alive       2015May\n",
      "243         17              sediment   27   <NA>       2015May\n",
      "244         18   Pocillopora_eydouxi   35   dead       2015May\n",
      "245         18              sediment   27   <NA>       2015May\n",
      "246         19              sediment    8   <NA>       2015May\n",
      "247         21   Pocillopora_eydouxi   35   gone       2015May\n",
      "248         21              sediment    8   <NA>       2015May\n",
      "249         22              sediment    8   <NA>       2015May\n",
      "250         23              sediment    8   <NA>       2015May\n",
      "251         24        Porites_lobata   35     UK       2015May\n",
      "252         24              sediment    8   <NA>       2015May\n",
      "253         25     Montipora_foliosa   35   dead       2015May\n",
      "254         27     Montipora_foliosa   35   dead       2015May\n",
      "255         28   Pocillopora_eydouxi   35   dead       2015May\n",
      "256         29     Montipora_foliosa   35   dead       2015May\n",
      "257         30     Montipora_foliosa   35   dead       2015May\n",
      "258         34   Pocillopora_eydouxi   35   dead       2015May\n",
      "259         35        Porites_lobata   35   gone       2015May\n",
      "260         36   Pocillopora_eydouxi   35   dead       2015May\n",
      "261         37   Pocillopora_eydouxi   30   dead       2015May\n",
      "262         40   Pocillopora_eydouxi   30  alive       2015May\n",
      "263         41        Porites_lobata   30     UK       2015May\n",
      "264         43     Montipora_foliosa   30   gone       2015May\n",
      "265         44   Pocillopora_eydouxi   30   gone       2015May\n",
      "266         45        Porites_lobata   30   gone       2015May\n",
      "267         46     Montipora_foliosa   30   gone       2015May\n",
      "268         47   Pocillopora_eydouxi   30     UK       2015May\n",
      "269         49        Porites_lobata   30   gone       2015May\n",
      "270         51   Pocillopora_eydouxi   30   gone       2015May\n",
      "271         52   Pocillopora_eydouxi   30   dead       2015May\n",
      "272         54        Porites_lobata   30  alive       2015May\n",
      "273         55   Pocillopora_eydouxi   30  alive       2015May\n",
      "274         56     Montipora_foliosa   30   dead       2015May\n",
      "275         58        Porites_lobata   30   gone       2015May\n",
      "276         59     Montipora_foliosa   30   gone       2015May\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "277         60   Pocillopora_eydouxi   30   dead       2015May\n",
      "278         62        Porites_lobata   30     UK       2015May\n",
      "279         63        Porites_lobata   30   gone       2015May\n",
      "280         64   Pocillopora_eydouxi   30     UK       2015May\n",
      "281         66     Montipora_foliosa   30   dead       2015May\n",
      "282         70     Montipora_foliosa   30   dead       2015May\n",
      "283         71   Pocillopora_eydouxi   30   dead       2015May\n",
      "284         73     Montipora_foliosa   30     UK       2015May\n",
      "285         74        Porites_lobata   30  alive       2015May\n",
      "286        128        Porites_lobata   30   gone       2015May\n",
      "287        129     Montipora_foliosa   30   gone       2015May\n",
      "288        130     Montipora_foliosa   30     UK       2015May\n",
      "289        133        Porites_lobata   30   gone       2015May\n",
      "290        134        Porites_lobata   27   gone       2015May\n",
      "291        135        Porites_lobata   27   gone       2015May\n",
      "292        137   Pocillopora_eydouxi   27   dead       2015May\n",
      "293        141        Porites_lobata   27  alive       2015May\n",
      "294        142     Montipora_foliosa   27     UK       2015May\n",
      "295        143   Pocillopora_eydouxi   27     UK       2015May\n",
      "296        144     Montipora_foliosa   27   gone       2015May\n",
      "297        145        Porites_lobata   27  alive       2015May\n",
      "298        146     Montipora_foliosa   27   gone       2015May\n",
      "299        150   Pocillopora_eydouxi   27   gone       2015May\n",
      "300        152     Montipora_foliosa   35   dead       2015May\n",
      "301        154   Pocillopora_eydouxi   35   dead       2015May\n",
      "302        155   Pocillopora_eydouxi   35     UK       2015May\n",
      "303        158     Montipora_foliosa   35   dead       2015May\n",
      "304        159        Porites_lobata   35   gone       2015May\n",
      "305        160   Pocillopora_eydouxi   35   gone       2015May\n",
      "306        161        Porites_lobata   35     UK       2015May\n",
      "307        165     Montipora_foliosa    8   dead       2015May\n",
      "308        168        Porites_lobata    8  alive       2015May\n",
      "309        170     Montipora_foliosa    8     UK       2015May\n",
      "310        172     Montipora_foliosa    8   dead       2015May\n",
      "311        173   Pocillopora_eydouxi    8   dead       2015May\n",
      "312        174     Montipora_foliosa    8   gone       2015May\n",
      "313        175   Pocillopora_eydouxi    8   gone       2015May\n",
      "314        176        Porites_lobata    8  alive       2015May\n",
      "315        180     Montipora_foliosa    8   dead       2015May\n",
      "316        181     Montipora_foliosa    8   dead       2015May\n",
      "317        182        Porites_lobata    8     UK       2015May\n",
      "318        183   Pocillopora_eydouxi    8   dead       2015May\n",
      "319        199     Montipora_foliosa   27   gone       2015May\n",
      "320        201        Porites_lobata   27  alive       2015May\n",
      "321        208     Montipora_foliosa   27   dead       2015May\n",
      "322        209   Pocillopora_eydouxi   27   gone       2015May\n",
      "323        210     Montipora_foliosa   27   gone       2015May\n",
      "324        211     Montipora_foliosa   27   gone       2015May\n",
      "325        250        Porites_lobata   34  alive       2015May\n",
      "326        251     Montipora_foliosa   34  alive       2015May\n",
      "327        252 Hydnophora_microconos   34     UK       2015May\n",
      "328        253     Montipora_foliosa   34  alive       2015May\n",
      "329        254        Porites_lobata   34   gone       2015May\n",
      "330        256     Montipora_foliosa   34   dead       2015May\n",
      "331        257   Pocillopora_eydouxi   34   dead       2015May\n",
      "332        258   Pocillopora_eydouxi   34   gone       2015May\n",
      "333        261              Favia_sp   34   gone       2015May\n",
      "334        262 Hydnophora_microconos   34   gone       2015May\n",
      "335        264        Porites_lobata   34  alive       2015May\n",
      "336        265   Pocillopora_eydouxi   34     UK       2015May\n",
      "337        266     Montipora_foliosa   34   dead       2015May\n",
      "338        267        Porites_lobata   34  alive       2015May\n",
      "339        268   Pocillopora_eydouxi   34   gone       2015May\n",
      "340        269        Porites_lobata   34  alive       2015May\n",
      "341        270   Pocillopora_eydouxi   34   dead       2015May\n",
      "342        271     Montipora_foliosa   34     UK       2015May\n",
      "343        272 Hydnophora_microconos   34  alive       2015May\n",
      "344        273              Favia_sp   34     UK       2015May\n",
      "345        274   Pocillopora_eydouxi   34   dead       2015May\n",
      "346        275 Hydnophora_microconos   34   gone       2015May\n",
      "347        276     Montipora_foliosa   34     UK       2015May\n",
      "348        278        Porites_lobata   34   gone       2015May\n",
      "349        280     Montipora_foliosa   34  alive       2015May\n",
      "350        281   Pocillopora_eydouxi   34   dead       2015May\n",
      "351        337     Montipora_foliosa   34     UK       2015May\n",
      "352        338        Porites_lobata   34  alive       2015May\n",
      "353        339   Pocillopora_eydouxi   34   dead       2015May\n",
      "354        341              Favia_sp   34   gone       2015May\n",
      "355        342 Hydnophora_microconos   34   dead       2015May\n",
      "356        343        Porites_lobata   34   gone       2015May\n",
      "357        344   Pocillopora_eydouxi   34   dead       2015May\n",
      "358        347              Favia_sp   34   dead       2015May\n",
      "359        350         Favia_matthai   34   gone       2015May\n",
      "360          1                 water   35   <NA>       2015May\n",
      "361          2                 water   35   <NA>       2015May\n",
      "362          3                 water   35   <NA>       2015May\n",
      "363          4                 water   35   <NA>       2015May\n",
      "364          5                 water   30   <NA>       2015May\n",
      "365          6                 water   30   <NA>       2015May\n",
      "366          7                 water   30   <NA>       2015May\n",
      "367          8                 water   30   <NA>       2015May\n",
      "368         17                 water   27   <NA>       2015May\n",
      "369         18                 water   27   <NA>       2015May\n",
      "370         19                 water   27   <NA>       2015May\n",
      "371         20                 water   27   <NA>       2015May\n",
      "372         23                 water    8   <NA>       2015May\n",
      "373         24                 water    8   <NA>       2015May\n",
      "374          7              sediment   27   <NA>      2015July\n",
      "375          8              sediment   27   <NA>      2015July\n",
      "376          9              sediment   27   <NA>      2015July\n",
      "377         39              sediment    8   <NA>      2015July\n",
      "378         40              sediment    8   <NA>      2015July\n",
      "379         42              sediment    8   <NA>      2015July\n",
      "380         51              sediment   32   <NA>      2015July\n",
      "381         52              sediment   32   <NA>      2015July\n",
      "382         54              sediment   32   <NA>      2015July\n",
      "                               ref\n",
      "1                      KI2014.tag1\n",
      "2                  KI2014.tag2_572\n",
      "3                      KI2014.tag3\n",
      "4           KI2014.sediment3.site8\n",
      "5           KI2014.sediment4.site8\n",
      "6                      KI2014.tag5\n",
      "7           KI2014.sediment5.site8\n",
      "8           KI2014.sediment6.site8\n",
      "9                      KI2014.tag7\n",
      "10          KI2014.sediment7.site8\n",
      "11                     KI2014.tag8\n",
      "12          KI2014.sediment8.site8\n",
      "13                     KI2014.tag9\n",
      "14        KI2014.sediment10.site35\n",
      "15        KI2014.sediment11.site35\n",
      "16                KI2014.tag12_635\n",
      "17        KI2014.sediment12.site35\n",
      "18                    KI2014.tag13\n",
      "19        KI2014.sediment13.site35\n",
      "20        KI2014.sediment14.site35\n",
      "21                    KI2014.tag15\n",
      "22        KI2014.sediment15.site30\n",
      "23        KI2014.sediment16.site30\n",
      "24        KI2014.sediment17.site30\n",
      "25        KI2014.sediment18.site30\n",
      "26        KI2014.sediment19.site30\n",
      "27        KI2014.sediment20.site30\n",
      "28                    KI2014.tag41\n",
      "29                    KI2014.tag43\n",
      "30                    KI2014.tag44\n",
      "31                    KI2014.tag50\n",
      "32        KI2014.sediment50.site35\n",
      "33        KI2014.sediment58.site27\n",
      "34        KI2014.sediment59.site27\n",
      "35                    KI2014.tag60\n",
      "36        KI2014.sediment60.site27\n",
      "37        KI2014.sediment61.site27\n",
      "38                    KI2014.tag63\n",
      "39        KI2014.sediment63.site27\n",
      "40                KI2014.tag69_875\n",
      "41                    KI2014.tag76\n",
      "42                    KI2014.tag84\n",
      "43                    KI2014.tag87\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "44                    KI2014.tag88\n",
      "45                KI2014.tag90_690\n",
      "46                KI2014.tag92_860\n",
      "47                    KI2014.tag94\n",
      "48                    KI2014.tag97\n",
      "49                    KI2014.tag98\n",
      "50                   KI2014.tag102\n",
      "51                   KI2014.tag105\n",
      "52                   KI2014.tag106\n",
      "53                   KI2014.tag107\n",
      "54                   KI2014.tag109\n",
      "55                   KI2014.tag126\n",
      "56                   KI2014.tag129\n",
      "57                   KI2014.tag131\n",
      "58                   KI2014.tag132\n",
      "59                   KI2014.tag133\n",
      "60                   KI2014.tag134\n",
      "61                   KI2014.tag135\n",
      "62                   KI2014.tag136\n",
      "63                   KI2014.tag137\n",
      "64                   KI2014.tag140\n",
      "65                   KI2014.tag142\n",
      "66                   KI2014.tag143\n",
      "67                   KI2014.tag144\n",
      "68                   KI2014.tag147\n",
      "69                   KI2014.tag149\n",
      "70                   KI2014.tag151\n",
      "71                   KI2014.tag265\n",
      "72                   KI2014.tag385\n",
      "73                   KI2014.tag387\n",
      "74                   KI2014.tag448\n",
      "75               KI2014.tag451_871\n",
      "76                   KI2014.tag453\n",
      "77                   KI2014.tag456\n",
      "78                   KI2014.tag467\n",
      "79                   KI2014.tag468\n",
      "80                   KI2014.tag469\n",
      "81                   KI2014.tag474\n",
      "82                   KI2014.tag476\n",
      "83            KI2014.water1.site35\n",
      "84             KI2014.water3.site8\n",
      "85             KI2014.water4.site8\n",
      "86            KI2014.water5.site35\n",
      "87            KI2014.water6.site35\n",
      "88            KI2014.water7.site35\n",
      "89            KI2014.water8.site30\n",
      "90           KI2014.water11.site27\n",
      "91           KI2014.water12.site27\n",
      "92           KI2014.water13.site27\n",
      "93           KI2014.water14.site27\n",
      "94           KI2014.water15.site30\n",
      "95           KI2014.water16.site30\n",
      "96           KI2014.water17.site30\n",
      "97            KI2014.water21.site8\n",
      "98           KI2014.water28.site35\n",
      "99           KI2014.water29.site35\n",
      "100          KI2014.water30.site35\n",
      "101           KI2014.water31.site8\n",
      "102          KI2014.water32.site30\n",
      "103               KI2015a_Pre.tag1\n",
      "104               KI2015a_Pre.tag3\n",
      "105              KI2015a_Pre.tag5b\n",
      "106           KI2015a_Pre.tag2_572\n",
      "107              KI2015a_Pre.tag6b\n",
      "108               KI2015a_Pre.tag7\n",
      "109              KI2015a_Pre.tag87\n",
      "110  KI2015a_Post.sediment8.site35\n",
      "111              KI2015a_Pre.tag14\n",
      "112  KI2015a_Post.sediment9.site35\n",
      "113               KI2015a_Pre.tag9\n",
      "114 KI2015a_Post.sediment10.site35\n",
      "115             KI2015a_Pre.tag10b\n",
      "116 KI2015a_Post.sediment11.site35\n",
      "117 KI2015a_Post.sediment12.site35\n",
      "118             KI2015a_Pre.tag574\n",
      "119 KI2015a_Post.sediment13.site30\n",
      "120 KI2015a_Post.sediment14.site30\n",
      "121 KI2015a_Post.sediment15.site30\n",
      "122              KI2015a_Pre.tag88\n",
      "123 KI2015a_Post.sediment16.site30\n",
      "124          KI2015a_Pre.tag90_690\n",
      "125 KI2015a_Post.sediment17.site30\n",
      "126 KI2015a_Post.sediment18.site30\n",
      "127             KI2015a_Pre.tag15b\n",
      "128 KI2015a_Post.sediment19.site27\n",
      "129 KI2015a_Post.sediment20.site27\n",
      "130          KI2015a_Pre.tag92_860\n",
      "131 KI2015a_Post.sediment21.site27\n",
      "132              KI2015a_Pre.tag94\n",
      "133 KI2015a_Post.sediment22.site27\n",
      "134 KI2015a_Post.sediment23.site27\n",
      "135 KI2015a_Post.sediment24.site27\n",
      "136  KI2015a_Post.sediment25.site8\n",
      "137  KI2015a_Post.sediment26.site8\n",
      "138  KI2015a_Post.sediment27.site8\n",
      "139  KI2015a_Post.sediment29.site8\n",
      "140  KI2015a_Post.sediment30.site8\n",
      "141              KI2015a_Post.tag1\n",
      "142          KI2015a_Post.tag2_572\n",
      "143              KI2015a_Post.tag3\n",
      "144              KI2015a_Post.tag5\n",
      "145            KI2015a_Post.tag632\n",
      "146              KI2015a_Post.tag8\n",
      "147             KI2015a_Post.tag13\n",
      "148            KI2015a_Post.tag634\n",
      "149         KI2015a_Post.tag12_635\n",
      "150             KI2015a_Post.tag15\n",
      "151            KI2015a_Post.tag636\n",
      "152             KI2015a_Post.tag88\n",
      "153         KI2015a_Post.tag92_860\n",
      "154             KI2015a_Post.tag97\n",
      "155             KI2015a_Post.tag94\n",
      "156             KI2015a_Post.tag98\n",
      "157            KI2015a_Post.tag102\n",
      "158            KI2015a_Post.tag106\n",
      "159            KI2015a_Post.tag105\n",
      "160            KI2015a_Post.tag107\n",
      "161            KI2015a_Post.tag109\n",
      "162             KI2015a_Post.tag50\n",
      "163             KI2015a_Post.tag60\n",
      "164             KI2015a_Post.tag63\n",
      "165         KI2015a_Post.tag69_875\n",
      "166             KI2015a_Post.tag76\n",
      "167             KI2015a_Post.tag84\n",
      "168           KI2015a_Post.tag126b\n",
      "169            KI2015a_Post.tag131\n",
      "170            KI2015a_Post.tag129\n",
      "171            KI2015a_Post.tag132\n",
      "172            KI2015a_Post.tag133\n",
      "173            KI2015a_Post.tag134\n",
      "174            KI2015a_Post.tag135\n",
      "175            KI2015a_Post.tag136\n",
      "176            KI2015a_Post.tag137\n",
      "177            KI2015a_Post.tag140\n",
      "178            KI2015a_Post.tag142\n",
      "179            KI2015a_Post.tag144\n",
      "180            KI2015a_Post.tag143\n",
      "181            KI2015a_Post.tag149\n",
      "182            KI2015a_Post.tag147\n",
      "183            KI2015a_Post.tag151\n",
      "184            KI2015a_Post.tag385\n",
      "185            KI2015a_Post.tag387\n",
      "186            KI2015a_Post.tag474\n",
      "187            KI2015a_Post.tag476\n",
      "188            KI2015a_Post.tag477\n",
      "189            KI2015a_Post.tag469\n",
      "190            KI2015a_Post.tag468\n",
      "191            KI2015a_Post.tag467\n",
      "192            KI2015a_Post.tag448\n",
      "193        KI2015a_Post.tag451_871\n",
      "194            KI2015a_Post.tag456\n",
      "195            KI2015a_Post.tag453\n",
      "196            KI2015a_Post.tag265\n",
      "197             KI2015a_Post.tag41\n",
      "198             KI2015a_Post.tag43\n",
      "199             KI2015a_Post.tag44\n",
      "200      KI2015a_Post.water5.site8\n",
      "201      KI2015a_Post.water6.site8\n",
      "202      KI2015a_Post.water7.site8\n",
      "203      KI2015a_Post.water8.site8\n",
      "204     KI2015a_Post.water9.site35\n",
      "205    KI2015a_Post.water10.site35\n",
      "206    KI2015a_Post.water11.site35\n",
      "207    KI2015a_Post.water12.site35\n",
      "208    KI2015a_Post.water13.site27\n",
      "209    KI2015a_Post.water14.site27\n",
      "210    KI2015a_Post.water15.site27\n",
      "211    KI2015a_Post.water16.site27\n",
      "212    KI2015a_Post.water17.site30\n",
      "213    KI2015a_Post.water18.site30\n",
      "214    KI2015a_Post.water19.site30\n",
      "215    KI2015a_Post.water20.site30\n",
      "216                   KI2015b.tag1\n",
      "217       KI2015b.sediment1.site35\n",
      "218       KI2015b.sediment2.site35\n",
      "219                   KI2015b.tag3\n",
      "220       KI2015b.sediment3.site35\n",
      "221               KI2015b.tag2_572\n",
      "222       KI2015b.sediment4.site35\n",
      "223                   KI2015b.tag5\n",
      "224       KI2015b.sediment5.site35\n",
      "225                   KI2015b.tag8\n",
      "226       KI2015b.sediment6.site35\n",
      "227       KI2015b.sediment7.site30\n",
      "228       KI2015b.sediment8.site30\n",
      "229                 KI2015b.tag634\n",
      "230       KI2015b.sediment9.site30\n",
      "231              KI2015b.tag12_635\n",
      "232      KI2015b.sediment10.site30\n",
      "233                 KI2015b.tag574\n",
      "234      KI2015b.sediment12.site30\n",
      "235                  KI2015b.tag87\n",
      "236      KI2015b.sediment13.site27\n",
      "237                  KI2015b.tag13\n",
      "238      KI2015b.sediment14.site27\n",
      "239      KI2015b.sediment15.site27\n",
      "240                 KI2015b.tag636\n",
      "241      KI2015b.sediment16.site27\n",
      "242                  KI2015b.tag15\n",
      "243      KI2015b.sediment17.site27\n",
      "244                  KI2015b.tag88\n",
      "245      KI2015b.sediment18.site27\n",
      "246       KI2015b.sediment19.site8\n",
      "247              KI2015b.tag92_860\n",
      "248       KI2015b.sediment21.site8\n",
      "249       KI2015b.sediment22.site8\n",
      "250       KI2015b.sediment23.site8\n",
      "251                  KI2015b.tag97\n",
      "252       KI2015b.sediment24.site8\n",
      "253                  KI2015b.tag98\n",
      "254                 KI2015b.tag102\n",
      "255                 KI2015b.tag106\n",
      "256                 KI2015b.tag105\n",
      "257                 KI2015b.tag110\n",
      "258                 KI2015b.tag116\n",
      "259                 KI2015b.tag119\n",
      "260                 KI2015b.tag120\n",
      "261                 KI2015b.tag132\n",
      "262                 KI2015b.tag129\n",
      "263                 KI2015b.tag131\n",
      "264                 KI2015b.tag134\n",
      "265                 KI2015b.tag135\n",
      "266                 KI2015b.tag136\n",
      "267                 KI2015b.tag137\n",
      "268                 KI2015b.tag139\n",
      "269                 KI2015b.tag144\n",
      "270                 KI2015b.tag145\n",
      "271                 KI2015b.tag149\n",
      "272                 KI2015b.tag151\n",
      "273                 KI2015b.tag152\n",
      "274                 KI2015b.tag385\n",
      "275                 KI2015b.tag387\n",
      "276                 KI2015b.tag428\n",
      "277                 KI2015b.tag427\n",
      "278                 KI2015b.tag430\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "279                 KI2015b.tag434\n",
      "280                 KI2015b.tag436\n",
      "281                 KI2015b.tag437\n",
      "282                 KI2015b.tag443\n",
      "283                 KI2015b.tag440\n",
      "284                 KI2015b.tag446\n",
      "285                 KI2015b.tag441\n",
      "286                 KI2015b.tag682\n",
      "287                 KI2015b.tag133\n",
      "288                 KI2015b.tag147\n",
      "289             KI2015b.tag445_683\n",
      "290                 KI2015b.tag476\n",
      "291                 KI2015b.tag474\n",
      "292                 KI2015b.tag477\n",
      "293             KI2015b.tag451_871\n",
      "294                 KI2015b.tag456\n",
      "295                 KI2015b.tag469\n",
      "296                  KI2015b.tag44\n",
      "297                  KI2015b.tag46\n",
      "298                 KI2015b.tag684\n",
      "299                 KI2015b.tag688\n",
      "300              KI2015b.tag90_690\n",
      "301                 KI2015b.tag109\n",
      "302             KI2015b.tag114_692\n",
      "303             KI2015b.tag121_693\n",
      "304                KI2015b.tag122b\n",
      "305                 KI2015b.tag123\n",
      "306             KI2015b.tag124_694\n",
      "307                  KI2015b.tag60\n",
      "308                  KI2015b.tag63\n",
      "309              KI2015b.tag69_875\n",
      "310                  KI2015b.tag76\n",
      "311                  KI2015b.tag84\n",
      "312                  KI2015b.tag85\n",
      "313                  KI2015b.tag86\n",
      "314                 KI2015b.tag241\n",
      "315             KI2015b.tag256_697\n",
      "316                 KI2015b.tag698\n",
      "317                 KI2015b.tag717\n",
      "318                 KI2015b.tag718\n",
      "319                 KI2015b.tag721\n",
      "320                 KI2015b.tag723\n",
      "321                 KI2015b.tag730\n",
      "322                 KI2015b.tag731\n",
      "323                 KI2015b.tag732\n",
      "324                 KI2015b.tag733\n",
      "325            KI2015b.tag314_1088\n",
      "326             KI2015b.tag317_949\n",
      "327                 KI2015b.tag323\n",
      "328                 KI2015b.tag324\n",
      "329                 KI2015b.tag325\n",
      "330                 KI2015b.tag329\n",
      "331                 KI2015b.tag331\n",
      "332                 KI2015b.tag336\n",
      "333                 KI2015b.tag348\n",
      "334                 KI2015b.tag349\n",
      "335                 KI2015b.tag354\n",
      "336                 KI2015b.tag355\n",
      "337                 KI2015b.tag363\n",
      "338                 KI2015b.tag361\n",
      "339                 KI2015b.tag359\n",
      "340                 KI2015b.tag773\n",
      "341                 KI2015b.tag774\n",
      "342                 KI2015b.tag775\n",
      "343                 KI2015b.tag776\n",
      "344                 KI2015b.tag777\n",
      "345                 KI2015b.tag778\n",
      "346                 KI2015b.tag779\n",
      "347                 KI2015b.tag780\n",
      "348                 KI2015b.tag782\n",
      "349                 KI2015b.tag784\n",
      "350                 KI2015b.tag785\n",
      "351                 KI2015b.tag841\n",
      "352                 KI2015b.tag842\n",
      "353                 KI2015b.tag843\n",
      "354                 KI2015b.tag845\n",
      "355                 KI2015b.tag846\n",
      "356                 KI2015b.tag847\n",
      "357                 KI2015b.tag848\n",
      "358                 KI2015b.tag851\n",
      "359                 KI2015b.tag854\n",
      "360          KI2015b.water1.site35\n",
      "361          KI2015b.water2.site35\n",
      "362          KI2015b.water3.site35\n",
      "363          KI2015b.water4.site35\n",
      "364          KI2015b.water5.site30\n",
      "365          KI2015b.water6.site30\n",
      "366          KI2015b.water7.site30\n",
      "367          KI2015b.water8.site30\n",
      "368         KI2015b.water17.site27\n",
      "369         KI2015b.water18.site27\n",
      "370         KI2015b.water19.site27\n",
      "371         KI2015b.water20.site27\n",
      "372          KI2015b.water23.site8\n",
      "373          KI2015b.water24.site8\n",
      "374       KI2015c.sediment7.site27\n",
      "375       KI2015c.sediment8.site27\n",
      "376       KI2015c.sediment9.site27\n",
      "377       KI2015c.sediment39.site8\n",
      "378       KI2015c.sediment40.site8\n",
      "379       KI2015c.sediment42.site8\n",
      "380      KI2015c.sediment51.site32\n",
      "381      KI2015c.sediment52.site32\n",
      "382      KI2015c.sediment54.site32\n",
      "> \n",
      "> map.compartment.forcat <- map.compartment[,c(1,3:5,7:9,11)]\n",
      "> \n",
      "> \n",
      "> metadata <- join_all(list(map.compartment.forcat,meta.forcat),by='ref',match='first')\n",
      "> names(metadata)\n",
      " [1] \"ref\"                       \"SampleID\"                 \n",
      " [3] \"coral_tag\"                 \"SampleType\"               \n",
      " [5] \"field_season\"              \"Coral_Species\"            \n",
      " [7] \"site\"                      \"Status\"                   \n",
      " [9] \"field_season\"              \"site\"                     \n",
      "[11] \"coral_tag\"                 \"before.after\"             \n",
      "[13] \"annotator\"                 \"date_annotated\"           \n",
      "[15] \"colony_width1\"             \"colony_width2\"            \n",
      "[17] \"colony_area\"               \"binary_tissue_loss\"       \n",
      "[19] \"binary_disease\"            \"cyanobacteria_contact\"    \n",
      "[21] \"binary_algal_contact\"      \"turf_contact\"             \n",
      "[23] \"binary_macroalgae_contact\" \"type_of_macroalgae\"       \n",
      "[25] \"binary_CCA_contact\"        \"sediment_contact\"         \n",
      "[27] \"binary_coral_contact\"      \"binary_bleaching\"         \n",
      "[29] \"bleaching_proportion\"      \"notes_bleaching\"          \n",
      "[31] \"notes\"                     \"Year_Pre_Post\"            \n",
      "> rownames(metadata) <- metadata[,1] # Make rownames from SampleID\n",
      "> # metadata<-subset(metadata,select=-c(SampleID)) # Remove SampleID column\n",
      "> \n",
      "> write.csv(metadata, file=\"data/Coralphoto__Metadata/KI_Compartment_metadata.csv\")\n",
      "> write.table(metadata, file=\"data/Coralphoto__Metadata/KI_Compartment_metadata.tsv\", quote=FALSE, sep=\"\\t\", col.names = NA)\n",
      "> \n"
     ]
    }
   ],
   "source": [
    "# Now extract metadata from full metadata spreadsheet\n",
    "# Using mapping_file.txt\n",
    "R --vanilla < data/Coralphoto__Metadata/process_coral_metadata.R \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 3.4.1 (2017-06-30) -- \"Single Candle\"\n",
      "Copyright (C) 2017 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "  Natural language support but running in an English locale\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> # Script to build phyloseq object\n",
      "> library(stringr)\n",
      "> library(phyloseq)\n",
      "> library(reshape2)\n",
      "> \n",
      "> args= c(\"data/Bioinf/clust/all_rep_set_rep_set_nw_tophits.tsv\", \n",
      "+         \"data/Coralphoto__Metadata/KI_Compartment_metadata.tsv\", \n",
      "+         \"data/Bioinf/clust/97_otus_bysample.tsv\", \n",
      "+         \"data/ITS2db_trimmed_notuniques_otus.txt\", \n",
      "+         \"analyses/KI_Compartment.RData\")\n",
      "> \n",
      "> # Get command line arguments\n",
      "> # args = commandArgs(trailingOnly=TRUE)\n",
      "> # If two arguments not provided, return an error\n",
      "> if (length(args) < 5) {\n",
      "+   stop(\"must specify:\n",
      "+        1: tax data (e.g., nw_tophits.tsv); \n",
      "+        2: sample data (e.g., mapping_file.txt); \n",
      "+        3: OTU table (.tsv); \n",
      "+        4: duplicate reference taxa names (e.g., ITS2db_trimmed_notuniques_otus.txt)\n",
      "+        5: output .RData filename\", call.=FALSE)\n",
      "+ }\n",
      "> \n",
      "> # Import taxonomic assignment data from nw\n",
      "> read.nw <- function(file) {\n",
      "+   nw <- read.table(file, quote=\"\", stringsAsFactors=FALSE)\n",
      "+   nw <- nw[order(nw$otu),]\n",
      "+   return(nw)\n",
      "+ }\n",
      "> \n",
      "> tax <- read.nw(args[1])\n",
      "> \n",
      "> # Deal with identical taxonomic assignments (because some reference sequences are not unique...)\n",
      "> # Create names for groups of identical sequences based on members of group...\n",
      "> ident <- readLines(args[4])\n",
      "> ident <- gsub(\"denovo[0-9]*\\t\", \"\", ident)\n",
      "> ident <- strsplit(ident, split=\"\\t\")\n",
      "> ident2 <- lapply(ident, str_match_all, pattern=\"[A-I]{1}[0-9]{1,3}.*[_/]\")\n",
      "> ident2 <- lapply(ident2, function(g) gsub(pattern=\"\\\\..*_$\", \"\", x=unlist(g)))\n",
      "> ident2 <- lapply(ident2, function(g) gsub(pattern=\"_$\", \"\", g))\n",
      "> ident2 <- lapply(ident2, function(g) unlist(strsplit(g, split=\"/\")))\n",
      "> subtypes <- lapply(ident2, function(x) levels(factor(unlist(x))))\n",
      "> subtypes <- lapply(subtypes, function(s) paste(paste(s, collapse=\"/\"), \"_multiple\", sep=\"\"))\n",
      "> names(ident) <- subtypes\n",
      "> ident <- melt(do.call(rbind, ident))\n",
      "Warning message:\n",
      "In (function (..., deparse.level = 1)  :\n",
      "  number of columns of result is not a multiple of vector length (arg 6)\n",
      "> ident <- unique(ident[order(ident[,1]), c(3,1)])\n",
      "> \n",
      "> # Replace any sequence name in taxonomy assignment that is a member of a group of identical sequences with the name of the group\n",
      "> collapse.idents <- function(df) {\n",
      "+   within(df, {\n",
      "+     for (i in 1:nrow(ident)) {\n",
      "+       hit <- gsub(ident[i,1], ident[i,2], hit)\n",
      "+     }\n",
      "+   })\n",
      "+ }\n",
      "> \n",
      "> tax <- collapse.idents(tax)\n",
      "> tax <- with(tax, tax[order(otu, -sim), ])\n",
      "> tax <- tax[!duplicated(tax$otu), ]\n",
      "> rownames(tax) <- tax$otu\n",
      "> \n",
      "> # Import sample data\n",
      "> sam <- sample_data(read.delim(args[2], sep=\"\\t\", header=T, row.names=1))\n",
      "> row.names(sam) <- sam$SampleID\n",
      ">   \n",
      "> # Import OTU tables\n",
      "> otu <- otu_table(read.table(args[3], header=T, check.names=F, row.names=1,\n",
      "+                               skip=1, sep=\"\\t\", comment.char=\"\"), taxa_are_rows=T)\n",
      "> \n",
      "> if(length(sample_names(sam)) != length(sample_names(otu))) {\n",
      "+   print(\"Warning: number of samples is not the same in sample data (mapping file) and otu table\")\n",
      "+   print(\"Warning: removing sample names that are present in the mapping file, but not the otu table\")\n",
      "+   sample_names(sam) <- sam$SampleID\n",
      "+   sam <- sam[which(sample_names(sam) %in% sample_names(otu)),]\n",
      "+ }\n",
      "> \n",
      "> # Build phyloseq objects\n",
      "> phy <- phyloseq(otu, tax_table(as.matrix(tax)), sam)\n",
      "> \n",
      "> # Save phyloseq object\n",
      "> save(phy, file=file.path(args[5]))\n",
      "> \n",
      "> \n"
     ]
    }
   ],
   "source": [
    "# From SymITS2 by Ross Cunning https://github.com/jrcunning/SymITS2\n",
    "# Build a phyloseq object\n",
    "R --vanilla < ~/SymITS2/build_phyloseq.R --args data/Bioinf/clust/all_rep_set_rep_set_nw_tophits.tsv data/Coralphoto__Metadata/KI_Compartment_metadata.tsv data/Bioinf/clust/97_otus_bysample.tsv data/ITS2db_trimmed_notuniques_otus.txt analyses/KI_Compartment.RData\n",
    "# Arguments are 1) nw_tophits - taxonomic assignment output from run_nw.R, 2) sample metadata in tsv format, 3) OTU table, 4) duplicate reference taxa names, 5) output filename\n",
    "# Creates analyses/KI_Compartment.RData"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 3.4.1 (2017-06-30) -- \"Single Candle\"\n",
      "Copyright (C) 2017 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "  Natural language support but running in an English locale\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> # FILTER OUT OTUS THAT ARE NOT SYMBIODINIUM\n",
      "> library(stringr)\n",
      "> library(Biostrings)\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, cbind, colMeans, colnames,\n",
      "    colSums, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, lengths, Map, mapply, match,\n",
      "    mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "Loading required package: stats4\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Loading required package: XVector\n",
      "\n",
      "Attaching package: ‘Biostrings’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    strsplit\n",
      "\n",
      "> library(phyloseq)\n",
      "\n",
      "Attaching package: ‘phyloseq’\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    distance\n",
      "\n",
      "> \n",
      "> # Get command line arguments\n",
      "> args = commandArgs(trailingOnly=TRUE)\n",
      "> # If four arguments not provided, return an error\n",
      "> if (length(args) < 4) {\n",
      "+   stop(\"must specify \n",
      "+        1: phyloseq object (.RData); \n",
      "+        2: sequences; \n",
      "+        3: output filename\n",
      "+        4: local nt blast database\", call.=FALSE)\n",
      "+ }\n",
      "> \n",
      "> # Get tax data from phyloseq object\n",
      "> load(args[1])\n",
      "> tax <- data.frame(tax_table(phy), stringsAsFactors=F)\n",
      "> \n",
      "> # Identify taxa with poor matches to reference database\n",
      "> poortax <- subset(tax, as.numeric(sim) < 90)$otu\n",
      "> \n",
      "> # Get corresponding sequences\n",
      "> seqs <- readDNAStringSet(args[2])\n",
      "> names(seqs) <- gsub(\" .*$\", \"\", names(seqs))\n",
      "> poorseqs <- seqs[which(names(seqs) %in% poortax),]\n",
      "> \n",
      "> poorseqsfile <- file.path(dirname(args[2]), \"poorseqs.fasta\")\n",
      "> writeXStringSet(poorseqs, filepath=poorseqsfile, append=FALSE,\n",
      "+                 compress=FALSE, compression_level=NA, format=\"fasta\")\n",
      "> \n",
      "> \n",
      "> \n",
      "> \n",
      "> # TODO: ADD MINIMUM EVAL OR OTHER FILTER ON BLAST SEARCH?\n",
      "> # Blast poor matching sequences to local NCBI nt databse, get top hit, write to file\n",
      "> outfile <- file.path(dirname(args[2]), \"poorseqs_blast_results.txt\")\n",
      "> system(paste(\"blastn -db\", args[4],\n",
      "+             \"-query\", poorseqsfile, \"-num_threads 100\",\n",
      "+             \"-outfmt '6 qseqid qcovs stitle sseqid' -max_target_seqs 1 | sort -u -k1,1 >\",\n",
      "+             outfile))\n",
      "BLAST Database error: No alias or index file found for nucleotide database [/media/danielle/CoralDrive/blastdb] in search path [/home/danielle/git_repos/KI_Compartment::]\n",
      "> \n",
      "> # Split poor-matching sequences into files with 100 seqs each for blasting to nt database\n",
      "> # poorseqs <- readDNAStringSet(\"~/Documents/Academia/HIMB/USVI/STJ2015/data/otus_97_bysample/poorseqs.fasta\")\n",
      "> # poorseqsbins <- split(poorseqs, cut(1:length(poorseqs), breaks=length(poorseqs) %/% 100 + 1, labels=seq(1,length(poorseqs) %/% 100 + 1)))\n",
      "> # sapply(names(poorseqsbins), function(x) {\n",
      "> #   writeXStringSet(poorseqsbins[[x]], filepath = file.path(dirname(args[2]), paste0(\"poorseqs\", x,\".fasta\")))\n",
      "> # })\n",
      "> #\n",
      "> # outfile <- file.path(dirname(args[2]), \"poorseqs_blast_results.txt\")\n",
      "> # system(paste(\"rm\", outfile))\n",
      "> # poorseqsfiles <- list.files(path=file.path(dirname(args[2])), pattern=\"^poorseqs[0-9]\", full.names=T)\n",
      "> # \n",
      "> # library(parallel) # Set up for parallel processing\n",
      "> # cl <- makeCluster(detectCores())  # Initiate cluster\n",
      "> # clusterEvalQ(cl, library(Biostrings))  # Make Biostrings library available to cluster\n",
      "> # clusterExport(cl, c(\"poorseqsfiles\", \"outfile\"))  # Export objects to cluster\n",
      "> # \n",
      "> # parSapply(cl, poorseqsfiles, function(x) {\n",
      "> #   system(paste(\"blastn -db\", args[4],\n",
      "> #                \"-query\", x,\n",
      "> #                \"-outfmt '6 qseqid qcovs stitle sseqid' -max_target_seqs 1 | sort -u -k1,1 >>\",\n",
      "> #                outfile))\n",
      "> # })\n",
      "> # \n",
      "> # stopCluster(cl)  # Stop cluster\n",
      "> \n",
      "> \n",
      "> # Read BLAST results\n",
      "> poorseqs_blast <- readLines(outfile)\n",
      "> # If the top hit from NCBI does not contain the string \"Symbiodinium\", then this sequence is assumed to not be Symbiodinium.\n",
      "> poorseqs_sym <- data.frame(otu=str_extract(poorseqs_blast, \"denovo[^\\t]*\"),\n",
      "+                            symbio=str_detect(poorseqs_blast, \"Symbiodinium\"),   # TRUE if Symbiodinium\n",
      "+                            stringsAsFactors = FALSE)\n",
      "> \n",
      "> symbio <- merge(data.frame(otu=poortax), poorseqs_sym, all=T)\n",
      "> symbio[is.na(symbio)] <- FALSE  # If NA then no BLAST hit found... assume NOT Symbiodinium\n",
      "> \n",
      "> # Remove OTUs that are not Symbiodinium\n",
      "> phy.f <- prune_taxa(!taxa_names(phy) %in% symbio[symbio$symbio==FALSE, \"otu\"], phy)\n",
      "> \n",
      "> # Save filtered phyloseq object\n",
      "> save(phy.f, file=args[3])\n",
      "> \n"
     ]
    }
   ],
   "source": [
    "# From SymITS2 by Ross Cunning https://github.com/jrcunning/SymITS2\n",
    "# Filter out sequences which do not blast to Symbiodnium\n",
    "R --vanilla < ~/SymITS2/filter_notsym.R --args analyses/KI_Compartment.RData data/Bioinf/clust/all_rep_set_rep_set.fasta analyses/KI_Compartment.RData /media/danielle/CoralDrive/blastdb\n",
    "# Output file is KI_Compartment.RData for downstream analysis (next step = analyses/KI_Compartment_filter_samples.R)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 3.4.1 (2017-06-30) -- \"Single Candle\"\n",
      "Copyright (C) 2017 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "  Natural language support but running in an English locale\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> # Exporting phyloseq objects\n",
      "> \n",
      "> load(\"analyses/KI_Compartment.RData\")\n",
      "> \n",
      "> # Write tax table for phylogenetically-informed diversity\n",
      "> write.table(data.frame(phyloseq::tax_table(phy.f)), \"data/tax_table.txt\", row.names=T, quote=F)\n",
      "> # Write otu table for phylogenetically-informed diversity\n",
      "> write.table(data.frame(phyloseq::otu_table(phy.f)), \"data/otu_table.txt\", row.names=T, quote=F)\n",
      "> \n"
     ]
    }
   ],
   "source": [
    "R --vanilla < data/Bioinf/export_physeq_objects.R\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Build the phylogenetic tree to add to phyloseq object\n",
    "\n",
    "#Originally from Moorea_Sym pipeline\n",
    "#data/Bioinf/clust/all_rep_set_rep_set.fasta = seqs\n",
    "#data/tax_table.txt = ISDs\n",
    "#column 9 is clade ID\n",
    "#awk '{print $1}' data/tax_table.txt = denovo name\n",
    "#awk '{print $8}' data/tax_table.txt = clade name\n",
    "\n",
    "#subset ID lists by clade\n",
    "awk '$9 ~ /^A/{ print $1; }' data/tax_table.txt > data/Bioinf/tree/A_tree_seq.ids\n",
    "awk '$9 ~ /^B/{ print $1; }' data/tax_table.txt > data/Bioinf/tree/B_tree_seq.ids\n",
    "awk '$9 ~ /^C/{ print $1; }' data/tax_table.txt > data/Bioinf/tree/C_tree_seq.ids\n",
    "awk '$9 ~ /^D/{ print $1; }' data/tax_table.txt > data/Bioinf/tree/D_tree_seq.ids\n",
    "awk '$9 ~ /^E/{ print $1; }' data/tax_table.txt > data/Bioinf/tree/E_tree_seq.ids\n",
    "awk '$9 ~ /^F/{ print $1; }' data/tax_table.txt > data/Bioinf/tree/F_tree_seq.ids\n",
    "awk '$9 ~ /^G/{ print $1; }' data/tax_table.txt > data/Bioinf/tree/G_tree_seq.ids\n",
    "awk '$9 ~ /^H/{ print $1; }' data/tax_table.txt > data/Bioinf/tree/H_tree_seq.ids\n",
    "awk '$9 ~ /^I/{ print $1; }' data/tax_table.txt > data/Bioinf/tree/I_tree_seq.ids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "56\n",
      "1\n",
      "350\n",
      "51\n",
      "66\n",
      "26\n",
      "7\n"
     ]
    }
   ],
   "source": [
    "#filter seqs by id list\n",
    "filter_fasta.py -f data/Bioinf/clust/all_rep_set_rep_set.fasta -s data/Bioinf/tree/A_tree_seq.ids -o data/Bioinf/tree/A_tree_seqs.fasta\n",
    "filter_fasta.py -f data/Bioinf/clust/all_rep_set_rep_set.fasta -s data/Bioinf/tree/B_tree_seq.ids -o data/Bioinf/tree/B_tree_seqs.fasta\n",
    "filter_fasta.py -f data/Bioinf/clust/all_rep_set_rep_set.fasta -s data/Bioinf/tree/C_tree_seq.ids -o data/Bioinf/tree/C_tree_seqs.fasta\n",
    "filter_fasta.py -f data/Bioinf/clust/all_rep_set_rep_set.fasta -s data/Bioinf/tree/D_tree_seq.ids -o data/Bioinf/tree/D_tree_seqs.fasta\n",
    "#filter_fasta.py -f data/Bioinf/clust/all_rep_set_rep_set.fasta -s data/Bioinf/tree/E_tree_seq.ids -o data/Bioinf/tree/E_tree_seqs.fasta\n",
    "filter_fasta.py -f data/Bioinf/clust/all_rep_set_rep_set.fasta -s data/Bioinf/tree/F_tree_seq.ids -o data/Bioinf/tree/F_tree_seqs.fasta\n",
    "filter_fasta.py -f data/Bioinf/clust/all_rep_set_rep_set.fasta -s data/Bioinf/tree/G_tree_seq.ids -o data/Bioinf/tree/G_tree_seqs.fasta\n",
    "#filter_fasta.py -f data/Bioinf/clust/all_rep_set_rep_set.fasta -s data/Bioinf/tree/H_tree_seq.ids -o data/Bioinf/tree/H_tree_seqs.fasta\n",
    "filter_fasta.py -f data/Bioinf/clust/all_rep_set_rep_set.fasta -s data/Bioinf/tree/I_tree_seq.ids -o data/Bioinf/tree/I_tree_seqs.fasta\n",
    "\n",
    "#check seq number\n",
    "grep -c \">\" data/Bioinf/tree/A_tree_seqs.fasta\n",
    "grep -c \">\" data/Bioinf/tree/B_tree_seqs.fasta\n",
    "grep -c \">\" data/Bioinf/tree/C_tree_seqs.fasta\n",
    "grep -c \">\" data/Bioinf/tree/D_tree_seqs.fasta\n",
    "#grep -c \">\" data/Bioinf/tree/E_tree_seqs.fasta\n",
    "grep -c \">\" data/Bioinf/tree/F_tree_seqs.fasta\n",
    "grep -c \">\" data/Bioinf/tree/G_tree_seqs.fasta\n",
    "#grep -c \">\" data/Bioinf/tree/H_tree_seqs.fasta\n",
    "grep -c \">\" data/Bioinf/tree/I_tree_seqs.fasta\n",
    "\n",
    "#align fasta files for each clade\n",
    "align_seqs.py -i data/Bioinf/tree/A_tree_seqs.fasta -m muscle -o data/Bioinf/tree/\n",
    "align_seqs.py -i data/Bioinf/tree/B_tree_seqs.fasta -m muscle -o data/Bioinf/tree/\n",
    "align_seqs.py -i data/Bioinf/tree/C_tree_seqs.fasta -m muscle -o data/Bioinf/tree/\n",
    "align_seqs.py -i data/Bioinf/tree/D_tree_seqs.fasta -m muscle -o data/Bioinf/tree/\n",
    "#align_seqs.py -i data/Bioinf/tree/E_tree_seqs.fasta -m muscle -o data/Bioinf/tree/\n",
    "align_seqs.py -i data/Bioinf/tree/F_tree_seqs.fasta -m muscle -o data/Bioinf/tree/\n",
    "align_seqs.py -i data/Bioinf/tree/G_tree_seqs.fasta -m muscle -o data/Bioinf/tree/\n",
    "#align_seqs.py -i data/Bioinf/tree/H_tree_seqs.fasta -m muscle -o data/Bioinf/tree/\n",
    "align_seqs.py -i data/Bioinf/tree/I_tree_seqs.fasta -m muscle -o data/Bioinf/tree/\n",
    "\n",
    "#remove extra header info from alignments\n",
    "sed 's/ .*//' data/Bioinf/tree/A_tree_seqs_aligned.fasta > data/Bioinf/tree/A_tree_seqs_aligned_clean.fasta\n",
    "sed 's/ .*//' data/Bioinf/tree/B_tree_seqs_aligned.fasta > data/Bioinf/tree/B_tree_seqs_aligned_clean.fasta\n",
    "sed 's/ .*//' data/Bioinf/tree/C_tree_seqs_aligned.fasta > data/Bioinf/tree/C_tree_seqs_aligned_clean.fasta\n",
    "sed 's/ .*//' data/Bioinf/tree/D_tree_seqs_aligned.fasta > data/Bioinf/tree/D_tree_seqs_aligned_clean.fasta\n",
    "#sed 's/ .*//' data/Bioinf/tree/E_tree_seqs_aligned.fasta > data/Bioinf/tree/E_tree_seqs_aligned_clean.fasta\n",
    "sed 's/ .*//' data/Bioinf/tree/F_tree_seqs_aligned.fasta > data/Bioinf/tree/F_tree_seqs_aligned_clean.fasta\n",
    "sed 's/ .*//' data/Bioinf/tree/G_tree_seqs_aligned.fasta > data/Bioinf/tree/G_tree_seqs_aligned_clean.fasta\n",
    "#sed 's/ .*//' data/Bioinf/tree/H_tree_seqs_aligned.fasta > data/Bioinf/tree/H_tree_seqs_aligned_clean.fasta\n",
    "sed 's/ .*//' data/Bioinf/tree/I_tree_seqs_aligned.fasta > data/Bioinf/tree/I_tree_seqs_aligned_clean.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 3.4.1 (2017-06-30) -- \"Single Candle\"\n",
      "Copyright (C) 2017 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "  Natural language support but running in an English locale\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> # Import Libraries\n",
      "> library(stringr)\n",
      "> library(reshape2)\n",
      "> library(phyloseq)\n",
      "> library(seqinr)\n",
      "> library(phangorn) \n",
      "Loading required package: ape\n",
      "\n",
      "Attaching package: ‘ape’\n",
      "\n",
      "The following objects are masked from ‘package:seqinr’:\n",
      "\n",
      "    as.alignment, consensus\n",
      "\n",
      "> library(caroline)\n",
      "> \n",
      "> # Clear workspace\n",
      "> rm(list=ls())\n",
      "> \n",
      "> # Load filtered RData object from output of filter_notsym.R script\n",
      "> load(\"analyses/KI_Compartment.RData\")\n",
      "> \n",
      "> # Filter OTUs by minimum count\n",
      "> # Set threshold count\n",
      "> n <- 5\n",
      "> # Identify OTUs below threshold count\n",
      "> taxa <- taxa_sums(phy.f)[which(taxa_sums(phy.f) >= n)]\n",
      "> # Remove taxa below threshold count\n",
      "> phy.f <- prune_taxa(names(taxa), phy.f)\n",
      "> \n",
      "> # Filter samples by minimum count\n",
      "> # Set threshold number of reads\n",
      "> sn <- 200\n",
      "> # Remove samples with fewer reads than threshold\n",
      "> phy.f <- prune_samples(sample_sums(phy.f)>=sn, phy.f)\n",
      "> \n",
      "> # Characterize sites by disturbance level\n",
      "> VeryHigh <- c(33,40,32,31,27,30,26)\n",
      "> High <- c(25,3,38,24)\n",
      "> HighMed <- c(9,34,35,8,14,6)\n",
      "> LowMed <- c(2,22,1,23)\n",
      "> Low <- c(7,13,12,4,36,5,37)\n",
      "> VeryLow <- c(10,21,11,20,16,15,39,19,18,17)\n",
      "> \n",
      "> sample_data(phy.f)$site <- factor(sample_data(phy.f)$site)\n",
      "> sample_data(phy.f)$Dist <- sample_data(phy.f)$site\n",
      "> \n",
      "> for (i in VeryHigh){\n",
      "+   if(i %in% unique(as.character(data.frame(sample_data(phy.f))$site)))\n",
      "+     (sample_data(phy.f)$Dist<- gsub (paste(\"\\\\<\",as.character(i),\"\\\\>\",sep=\"\"),as.character(\"VeryHigh\"), as.character(data.frame(sample_data(phy.f))$Dist), as.character(\"VeryHigh\")))\n",
      "+ }\n",
      "> \n",
      "> for (i in High){\n",
      "+   if(i %in% unique(as.character(data.frame(sample_data(phy.f))$site)))\n",
      "+     (sample_data(phy.f)$Dist<- gsub (paste(\"\\\\<\",as.character(i),\"\\\\>\",sep=\"\"),as.character(\"High\"), as.character(data.frame(sample_data(phy.f))$Dist), as.character(\"High\")))\n",
      "+ }\n",
      "> \n",
      "> for (i in HighMed){\n",
      "+   if(i %in% unique(as.character(data.frame(sample_data(phy.f))$site)))\n",
      "+     (sample_data(phy.f)$Dist<- gsub (paste(\"\\\\<\",as.character(i),\"\\\\>\",sep=\"\"),as.character(\"HighMed\"), as.character(data.frame(sample_data(phy.f))$Dist), as.character(\"HighMed\")))\n",
      "+ }\n",
      "> \n",
      "> for (i in LowMed){\n",
      "+   if(i %in% unique(as.character(data.frame(sample_data(phy.f))$site)))\n",
      "+     (sample_data(phy.f)$Dist<- gsub (paste(\"\\\\<\",as.character(i),\"\\\\>\",sep=\"\"),as.character(\"LowMed\"), as.character(data.frame(sample_data(phy.f))$Dist), as.character(\"LowMed\")))\n",
      "+ }\n",
      "> \n",
      "> for (i in Low){\n",
      "+   if(i %in% unique(as.character(data.frame(sample_data(phy.f))$site)))\n",
      "+     (sample_data(phy.f)$Dist<- gsub (paste(\"\\\\<\",as.character(i),\"\\\\>\",sep=\"\"),as.character(\"Low\"), as.character(data.frame(sample_data(phy.f))$Dist), as.character(\"Low\")))\n",
      "+ }\n",
      "> \n",
      "> for (i in VeryLow){\n",
      "+   if(i %in% unique(as.character(data.frame(sample_data(phy.f))$site)))\n",
      "+     (sample_data(phy.f)$Dist<- gsub (paste(\"\\\\<\",as.character(i),\"\\\\>\",sep=\"\"),as.character(\"VeryLow\"), as.character(data.frame(sample_data(phy.f))$Dist), as.character(\"VeryLow\")))\n",
      "+ }\n",
      "> \n",
      "> levels(sample_data(phy.f)$Dist) <- c(\"VeryHigh\",\"High\",\"HighMed\",\"Low\",\"VeryLow\")\n",
      "> \n",
      "> # Assign new name for clarity\n",
      "> phy97.f.c <- phy.f\n",
      "> \n",
      "> # Make a tax_table column for \"clade\"\n",
      "> # Rename rank_names (aka column names in phy97.f.c)\n",
      "> colnames(tax_table(phy97.f.c)) <- c(otu = \"otu\", sim = \"sim\", del = \"del\", ins = \"ins\", mis = \"mis\", len = \"len\", score = \"score\", hit = \"hit\", i = \"clade\")\n",
      "> # Copy \"hit\" into last column\n",
      "> tax_table(phy97.f.c)[,9] <- tax_table(phy97.f.c)[,8]\n",
      "> \n",
      "> # Use gsub to replace full \"hit\" name with only clade letter\n",
      "> tax_table(phy97.f.c)[,9] <- gsub(\"^A.*\", \"A\", tax_table(phy97.f.c)[,9])\n",
      "> tax_table(phy97.f.c)[,9] <- gsub(\"^B.*\", \"B\", tax_table(phy97.f.c)[,9])\n",
      "> tax_table(phy97.f.c)[,9] <- gsub(\"^C.*\", \"C\", tax_table(phy97.f.c)[,9])\n",
      "> tax_table(phy97.f.c)[,9] <- gsub(\"^D.*\", \"D\", tax_table(phy97.f.c)[,9])\n",
      "> tax_table(phy97.f.c)[,9] <- gsub(\"^F.*\", \"F\", tax_table(phy97.f.c)[,9])\n",
      "> tax_table(phy97.f.c)[,9] <- gsub(\"^G.*\", \"G\", tax_table(phy97.f.c)[,9])\n",
      "> tax_table(phy97.f.c)[,9] <- gsub(\"^I.*\", \"I\", tax_table(phy97.f.c)[,9])\n",
      "> \n",
      "> # Write tax table for phylogenetically-informed diversity\n",
      "> write.table(data.frame(tax_table(phy97.f.c)), \"data/tax_table.txt\", row.names=T, quote=F)\n",
      "> # Write otu table for phylogenetically-informed diversity\n",
      "> write.delim(data.frame(otu_table(phy97.f.c)), \"data/otu_table.tsv\", quote = FALSE, row.names = T, sep = \"\\t\")\n",
      "> \n",
      "> #https://rdrr.io/rforge/seqinr/man/dist.alignment.html\n",
      "> #returns sqrt of pairwise genetic distance, then squared the matrices\n",
      "> A.seqs <- read.alignment(file = \"data/Bioinf/tree/A_tree_seqs_aligned_clean.fasta\", format= \"fasta\")\n",
      "> A.dis <- (as.matrix(dist.alignment(A.seqs, matrix = \"identity\" )))^2\n",
      "> write.csv(A.dis, file=\"data/Bioinf/tree/A.dis.matx.csv\")\n",
      "> \n",
      "> B.seqs <- read.alignment(file = \"data/Bioinf/tree/B_tree_seqs_aligned_clean.fasta\", format= \"fasta\")\n",
      "> B.dis <- (as.matrix(dist.alignment(B.seqs, matrix = \"identity\" )))^2\n",
      "> write.csv(B.dis, file=\"data/Bioinf/tree/B.dis.matx.csv\")\n",
      "> \n",
      "> C.seqs <- read.alignment(file = \"data/Bioinf/tree/C_tree_seqs_aligned_clean.fasta\", format= \"fasta\")\n",
      "> C.dis <- (as.matrix(dist.alignment(C.seqs, matrix = \"identity\" )))^2\n",
      "> write.csv(C.dis, file=\"data/Bioinf/tree/C.dis.matx.csv\")\n",
      "> \n",
      "> D.seqs <- read.alignment(file = \"data/Bioinf/tree/D_tree_seqs_aligned_clean.fasta\", format= \"fasta\")\n",
      "> D.dis <- (as.matrix(dist.alignment(D.seqs, matrix = \"identity\" )))^2\n",
      "> write.csv(D.dis, file=\"data/Bioinf/tree/D.dis.matx.csv\")\n",
      "> \n",
      "> F.seqs <- read.alignment(file = \"data/Bioinf/tree/F_tree_seqs_aligned_clean.fasta\", format= \"fasta\")\n",
      "> F.dis <- (as.matrix(dist.alignment(F.seqs, matrix = \"identity\" )))^2\n",
      "> write.csv(F.dis, file=\"data/Bioinf/tree/F.dis.matx.csv\")\n",
      "> \n",
      "> G.seqs <- read.alignment(file = \"data/Bioinf/tree/G_tree_seqs_aligned_clean.fasta\", format= \"fasta\")\n",
      "> G.dis <- (as.matrix(dist.alignment(G.seqs, matrix = \"identity\" )))^2\n",
      "> write.csv(G.dis, file=\"data/Bioinf/tree/G.dis.matx.csv\")\n",
      "> \n",
      "> I.seqs <- read.alignment(file = \"data/Bioinf/tree/I_tree_seqs_aligned_clean.fasta\", format= \"fasta\")\n",
      "> I.dis <- (as.matrix(dist.alignment(I.seqs, matrix = \"identity\" )))^2\n",
      "> write.csv(I.dis, file=\"data/Bioinf/tree/I.dis.matx.csv\")\n",
      "> \n",
      "> #give clade distances using average 28s distance from Pochon and Gates 2010\n",
      "> A_B <- matrix(0.219, ncol=ncol(A.dis), nrow=nrow(B.dis), dimnames=list(rownames(B.dis), colnames(A.dis)))\n",
      "> A_C <- matrix(0.1960, ncol=ncol(A.dis), nrow=nrow(C.dis), dimnames=list(rownames(C.dis), colnames(A.dis)))\n",
      "> A_D <- matrix(0.1775, ncol=ncol(A.dis), nrow=nrow(D.dis), dimnames=list(rownames(D.dis), colnames(A.dis)))\n",
      "> A_F <- matrix(0.2085, ncol=ncol(A.dis), nrow=nrow(F.dis), dimnames=list(rownames(F.dis), colnames(A.dis)))\n",
      "> A_G <- matrix(0.216, ncol=ncol(A.dis), nrow=nrow(G.dis), dimnames=list(rownames(G.dis), colnames(A.dis)))\n",
      "> A_I <- matrix(0.205, ncol=ncol(A.dis), nrow=nrow(I.dis), dimnames=list(rownames(I.dis), colnames(A.dis)))\n",
      "> B_C <- matrix(0.114, ncol=ncol(B.dis), nrow=nrow(C.dis), dimnames=list(rownames(C.dis), colnames(B.dis)))\n",
      "> B_D <- matrix(0.1705, ncol=ncol(B.dis), nrow=nrow(D.dis), dimnames=list(rownames(D.dis), colnames(B.dis)))\n",
      "> B_F <- matrix(0.1355, ncol=ncol(B.dis), nrow=nrow(F.dis), dimnames=list(rownames(F.dis), colnames(B.dis)))\n",
      "> B_G <- matrix(0.21, ncol=ncol(B.dis), nrow=nrow(G.dis), dimnames=list(rownames(G.dis), colnames(B.dis)))\n",
      "> B_I <- matrix(0.151, ncol=ncol(B.dis), nrow=nrow(I.dis), dimnames=list(rownames(I.dis), colnames(B.dis)))\n",
      "> C_D <- matrix(0.1520, ncol=ncol(C.dis), nrow=nrow(D.dis), dimnames=list(rownames(D.dis), colnames(C.dis)))\n",
      "> C_F <- matrix(0.0945, ncol=ncol(C.dis), nrow=nrow(F.dis), dimnames=list(rownames(F.dis), colnames(C.dis)))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> C_G <- matrix(0.187, ncol=ncol(C.dis), nrow=nrow(G.dis), dimnames=list(rownames(G.dis), colnames(C.dis)))\n",
      "> C_I <- matrix(0.137, ncol=ncol(C.dis), nrow=nrow(I.dis), dimnames=list(rownames(I.dis), colnames(C.dis)))\n",
      "> D_F <- matrix(0.1691, ncol=ncol(D.dis), nrow=nrow(F.dis), dimnames=list(rownames(F.dis), colnames(D.dis)))\n",
      "> D_G <- matrix(0.1795, ncol=ncol(D.dis), nrow=nrow(G.dis), dimnames=list(rownames(G.dis), colnames(D.dis)))\n",
      "> D_I <- matrix(0.169125, ncol=ncol(D.dis), nrow=nrow(I.dis), dimnames=list(rownames(I.dis), colnames(D.dis)))\n",
      "> F_G <- matrix(0.2072, ncol=ncol(F.dis), nrow=nrow(G.dis), dimnames=list(rownames(G.dis), colnames(F.dis)))\n",
      "> F_I <- matrix(0.14925, ncol=ncol(F.dis), nrow=nrow(I.dis), dimnames=list(rownames(I.dis), colnames(F.dis)))\n",
      "> G_I <- matrix(0.194, ncol=ncol(G.dis), nrow=nrow(I.dis), dimnames=list(rownames(I.dis), colnames(G.dis)))\n",
      "> \n",
      "> #build ACDG matrix\n",
      "> col1 <- rbind(A.dis, A_B, A_C, A_D, A_F, A_G, A_I)\n",
      "> col2 <- rbind(matrix(NA, nrow=nrow(A.dis), ncol=ncol(B.dis), dimnames=list(rownames(A.dis), colnames(B.dis))), B.dis, B_C, B_D, B_F, B_G, B_I)\n",
      "> col3 <- rbind(matrix(NA, nrow=nrow(A.dis)+nrow(B.dis), ncol=ncol(C.dis), dimnames=list(c(rownames(A.dis), rownames(B.dis)), colnames(C.dis))), C.dis, C_D, C_F, C_G, C_I)\n",
      "> col4 <- rbind(matrix(NA, nrow=nrow(A.dis)+nrow(B.dis)+nrow(C.dis), ncol=ncol(D.dis), dimnames=list(c(rownames(A.dis), rownames(B.dis), rownames(C.dis)), colnames(D.dis))), D.dis, D_F, D_G, D_I)\n",
      "> col5 <- rbind(matrix(NA, nrow=nrow(A.dis)+nrow(B.dis)+nrow(C.dis)+nrow(D.dis), ncol=ncol(F.dis), dimnames=list(c(rownames(A.dis), rownames(B.dis), rownames(C.dis), rownames(D.dis)), colnames(F.dis))), F.dis, F_G, F_I)\n",
      "> col6 <- rbind(matrix(NA, nrow=nrow(A.dis)+nrow(B.dis)+nrow(C.dis)+nrow(D.dis)+nrow(F.dis), ncol=ncol(G.dis), dimnames=list(c(rownames(A.dis), rownames(B.dis), rownames(C.dis), rownames(D.dis),  rownames(F.dis)), colnames(G.dis))), G.dis, G_I)\n",
      "> col7 <- rbind(matrix(NA, nrow=nrow(A.dis)+nrow(B.dis)+nrow(C.dis)+nrow(D.dis)+nrow(F.dis)+nrow(G.dis), ncol=ncol(I.dis), dimnames=list(c(rownames(A.dis), rownames(B.dis), rownames(C.dis), rownames(D.dis),  rownames(F.dis), rownames(G.dis)), colnames(I.dis))), I.dis)\n",
      "> \n",
      "> ubermatrix <- cbind(col1, col2, col3, col4, col5, col6, col7)\n",
      "> dim(ubermatrix)\n",
      "[1] 557 557\n",
      "> \n",
      "> #build tree\n",
      "> uber.tree <- phangorn::upgma(ubermatrix)\n",
      "> plot(uber.tree, main=\"UPGMA\")\n",
      "> \n",
      "> #write tree to file\n",
      "> write.tree(uber.tree, file=\"data/Bioinf/tree/uber.tre\")\n",
      "> \n",
      "> #use tree and OTU table for calculating beta_diversity.py\n",
      "> #http://qiime.org/scripts/beta_diversity.html\n",
      "> \n",
      "> class(uber.tree)\n",
      "[1] \"phylo\"\n",
      "> \n",
      "> phy_tree(uber.tree)\n",
      "\n",
      "Phylogenetic tree with 557 tips and 556 internal nodes.\n",
      "\n",
      "Tip labels:\n",
      "\tdenovo40263, denovo11160, denovo24565, denovo33967, denovo15150, denovo3197, ...\n",
      "\n",
      "Rooted; includes branch lengths.\n",
      "> \n",
      "> # Slot uber tree into the phy_tree slot of the phyloseq object\n",
      "> phy_tree(phy97.f.c) <- phy_tree(uber.tree)\n",
      "> \n",
      "> # Transform sample counts to proportional abundance for downstream analyses\n",
      "> phy97.f.c.p <- transform_sample_counts(phy97.f.c, function(x) x/sum(x))\n",
      "> \n",
      "> # Note, this is now the same as phy97.f.c - renaming and subsetting was completed before when water and sediment samples were included\n",
      "> phy97.f.c.coral <- subset_samples(phy97.f.c,SampleType==\"coral\")\n",
      "> phy97.f.c.coral <- subset_taxa(phy97.f.c.coral, taxa_sums(phy97.f.c.coral) > 0, prune=TRUE)\n",
      "> \n",
      "> phy97.f.c.water <- subset_samples(phy97.f.c,SampleType==\"water\")\n",
      "> phy97.f.c.water <- subset_taxa(phy97.f.c.water, taxa_sums(phy97.f.c.water) > 0, prune=TRUE)\n",
      "> \n",
      "> phy97.f.c.sediment <- subset_samples(phy97.f.c,SampleType==\"sediment\")\n",
      "> phy97.f.c.sediment <- subset_taxa(phy97.f.c.sediment, taxa_sums(phy97.f.c.sediment) > 0, prune=TRUE)\n",
      "> \n",
      "> # Transform sample counts to proportional abundance for downstream analyses\n",
      "> phy97.f.c.coral.p <- transform_sample_counts(phy97.f.c.coral, function(x) x/sum(x))\n",
      "> phy97.f.c.water.p <- transform_sample_counts(phy97.f.c.water, function(x) x/sum(x))\n",
      "> phy97.f.c.sediment.p <- transform_sample_counts(phy97.f.c.sediment, function(x) x/sum(x))\n",
      "> \n",
      "> # Calculate number of sequences in phy97.f.c\n",
      "> total_seqs <- sum(taxa_sums(phy97.f.c))\n",
      "> coral_seqs <- sum(taxa_sums(phy97.f.c.coral))\n",
      "> water_seqs <- sum(taxa_sums(phy97.f.c.water))\n",
      "> sediment_seqs <- sum(taxa_sums(phy97.f.c.sediment))\n",
      "> \n",
      "> # Cleanup \n",
      "> rm(VeryHigh,VeryLow,phy.f,Low,LowMed,High,HighMed)\n",
      "> \n",
      "> # Save grouped data as RData file\n",
      "> save(list=ls(pattern=\"phy97.f.c.\"), sediment_seqs, total_seqs, coral_seqs, water_seqs, file = \"data/KI_Compartment_f_coral_grouped.RData\")\n",
      "> \n"
     ]
    }
   ],
   "source": [
    "# Format and filter the RData file\n",
    "R --vanilla < analyses/KI_Compartment_filter_samples.R\n",
    "# Input file is analyses/KI_Compartment.RData\n",
    "# Must run build_phy_tree first for this to work properly\n",
    "# Output file is: data/KI_Compartment_f_coral_grouped.RData"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
